diff --git i/Rendering/OpenGL/vtkOpenGLPolyDataMapper.cxx w/Rendering/OpenGL/vtkOpenGLPolyDataMapper.cxx
index 036c37a4..0f6d5d46 100644
--- i/Rendering/OpenGL/vtkOpenGLPolyDataMapper.cxx
+++ w/Rendering/OpenGL/vtkOpenGLPolyDataMapper.cxx
@@ -65,8 +65,10 @@ vtkOpenGLPolyDataMapper::vtkOpenGLPolyDataMapper()
   this->TotalCells = 0;
   this->InternalColorTexture = 0;
 
-  VTK_LEGACY_BODY(vtkOpenGLPolyDataMapper::vtkOpenGLPolyDataMapper,
-    "VTK 6.2");
+  // Commenting out in FME repository since it posts many warnings in MacOS 
+  // binding which do not help anybody and which will disappear automatically on
+  // next update.
+  //VTK_LEGACY_BODY(vtkOpenGLPolyDataMapper::vtkOpenGLPolyDataMapper, "VTK 6.2");
 }
 
 // Destructor (don't call ReleaseGraphicsResources() since it is virtual
diff --git i/Rendering/Volume/vtkVolumeTextureMapper3D.cxx w/Rendering/Volume/vtkVolumeTextureMapper3D.cxx
index e2fa2a42..49dc8d59 100644
--- i/Rendering/Volume/vtkVolumeTextureMapper3D.cxx
+++ w/Rendering/Volume/vtkVolumeTextureMapper3D.cxx
@@ -679,7 +679,10 @@ vtkVolumeTextureMapper3D::vtkVolumeTextureMapper3D()
   this->UseCompressedTexture          = false;
   this->SupportsNonPowerOfTwoTextures = false;
 
-  VTK_LEGACY_BODY(vtkVolumeTextureMapper3D::vtkVolumeTextureMapper3D,"VTK 7.0");
+  // Commenting out in FME repository since it posts many warnings in MacOS 
+  // binding which do not help anybody and which will disappear automatically on
+  // next update.
+  //VTK_LEGACY_BODY(vtkVolumeTextureMapper3D::vtkVolumeTextureMapper3D,"VTK 7.0");
 }
 
 //-----------------------------------------------------------------------------
diff --git i/Rendering/VolumeOpenGL/vtkOpenGLVolumeTextureMapper3D.cxx w/Rendering/VolumeOpenGL/vtkOpenGLVolumeTextureMapper3D.cxx
index fea7904e..6b04adb3 100644
--- i/Rendering/VolumeOpenGL/vtkOpenGLVolumeTextureMapper3D.cxx
+++ w/Rendering/VolumeOpenGL/vtkOpenGLVolumeTextureMapper3D.cxx
@@ -54,7 +54,10 @@ vtkOpenGLVolumeTextureMapper3D::vtkOpenGLVolumeTextureMapper3D()
   this->AlphaLookupIndex             =  0;
   this->RenderWindow                 = NULL;
   this->SupportsCompressedTexture    = false;
-  VTK_LEGACY_BODY(vtkOpenGLVolumeTextureMapper3D::vtkOpenGLVolumeTextureMapper3D,"VTK 7.0");
+  // Commenting out in FME repository since it posts many warnings in MacOS 
+  // binding which do not help anybody and which will disappear automatically on
+  // next update.
+  //VTK_LEGACY_BODY(vtkOpenGLVolumeTextureMapper3D::vtkOpenGLVolumeTextureMapper3D,"VTK 7.0");
 }
 
 vtkOpenGLVolumeTextureMapper3D::~vtkOpenGLVolumeTextureMapper3D()
