Index: Domains/Chemistry/vtkBlueObeliskDataParser.cxx
===================================================================
--- Domains/Chemistry/vtkBlueObeliskDataParser.cxx	(revision 3980)
+++ Domains/Chemistry/vtkBlueObeliskDataParser.cxx	(revision 3982)
@@ -51,6 +51,13 @@
     CurrentElectronicConfiguration(new vtkStdString),
     CurrentFamily(new vtkStdString)
 {
+  std::string altElementsPath;
+  char* envVar(getenv("MLAB_FMEwork_ThirdParty"));
+  if (envVar){
+    std::string FMEworkThirdPartyPath(envVar);
+    altElementsPath = FMEworkThirdPartyPath + "/Sources/vtk7/share/vtk-7.1/vtkDomainsChemistry/elements.xml";
+  }
+
   // Find elements.xml. Check the share directory first, then the source dir
   // if running from the build directory before installing.
   struct stat buf;
@@ -62,10 +69,14 @@
     {
     this->SetFileName(VTK_BODR_DATA_PATH_BUILD "/elements.xml");
     }
+  else if (!altElementsPath.empty() && (stat(altElementsPath.c_str(), &buf) == 0))
+    {
+    this->SetFileName(altElementsPath.c_str());
+    }
   else
     {
     vtkErrorMacro(<<"Cannot find elements.xml. Checked " VTK_BODR_DATA_PATH
-                  " and " VTK_BODR_DATA_PATH_BUILD)
+                  " and " VTK_BODR_DATA_PATH_BUILD " and " << altElementsPath.c_str() )
     }
 }
 
