diff --git i/ThirdParty/hdf5/vtkhdf5/src/H5public.h w/ThirdParty/hdf5/vtkhdf5/src/H5public.h
index 929cb936..09c0b5ec 100644
--- i/ThirdParty/hdf5/vtkhdf5/src/H5public.h
+++ w/ThirdParty/hdf5/vtkhdf5/src/H5public.h
@@ -148,23 +148,51 @@ typedef int htri_t;
 
 /* Define the ssize_t type if it not is defined */
 #if H5_SIZEOF_SSIZE_T==0
-/* Undefine this size, we will re-define it in one of the sections below */
-#undef H5_SIZEOF_SSIZE_T
-#if H5_SIZEOF_SIZE_T==H5_SIZEOF_INT
-#ifdef WIN32
-#undef ssize_t
-#endif
-typedef int ssize_t;
-#       define H5_SIZEOF_SSIZE_T H5_SIZEOF_INT
-#elif H5_SIZEOF_SIZE_T==H5_SIZEOF_LONG
-typedef long ssize_t;
-#       define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG
-#elif H5_SIZEOF_SIZE_T==H5_SIZEOF_LONG_LONG
-typedef long long ssize_t;
-#       define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG_LONG
-#else /* Can't find matching type for ssize_t */
-#   error "nothing appropriate for ssize_t"
+
+  /* Undefine this size, we will re-define it in one of the sections below */
+  #undef H5_SIZEOF_SSIZE_T
+
+#if !defined(_WIN64) && (_MSC_VER < 1700)
+
+  // Annoyingly H5public typedefs the ssize_t which, however, on VC10-32, can appear elsewhere as 
+  // typedef which then causes collisions. Therefore we replace the typedef with a define which then
+  // can replace/coexist with the other type.
+  #if H5_SIZEOF_SIZE_T==H5_SIZEOF_INT
+    #if defined(_WIN32) && defined(ssize_t)
+      #undef ssize_t
+    #endif
+    #       define ssize_t int
+    #       define H5_SIZEOF_SSIZE_T H5_SIZEOF_INT
+  #elif H5_SIZEOF_SIZE_T==H5_SIZEOF_LONG
+    #       define ssize_t long
+    #       define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG
+  #elif H5_SIZEOF_SIZE_T==H5_SIZEOF_LONG_LONG
+    #       define ssize_t long long
+    #       define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG_LONG
+  #else /* Can't find matching type for ssize_t */
+    #   error "nothing appropriate for ssize_t"
+  #endif
+
+#else
+
+  #if H5_SIZEOF_SIZE_T==H5_SIZEOF_INT
+    #if defined(_WIN32) && defined(ssize_t)
+      #undef ssize_t
+    #endif
+    typedef int ssize_t;
+    #       define H5_SIZEOF_SSIZE_T H5_SIZEOF_INT
+  #elif H5_SIZEOF_SIZE_T==H5_SIZEOF_LONG
+    typedef long ssize_t;
+    #       define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG
+  #elif H5_SIZEOF_SIZE_T==H5_SIZEOF_LONG_LONG
+    typedef long long ssize_t;
+    #       define H5_SIZEOF_SSIZE_T H5_SIZEOF_LONG_LONG
+  #else /* Can't find matching type for ssize_t */
+    #   error "nothing appropriate for ssize_t"
+  #endif
+
 #endif
+
 #endif
 
 /*
