diff --git i/configure w/configure
index 1ec53dc..1961b6a 100644
--- i/configure
+++ w/configure
@@ -15258,8 +15258,14 @@ darwin* | rhapsody*)
   version_type=darwin
   need_lib_prefix=no
   need_version=no
-  library_names_spec='$libname$release$major$shared_ext $libname$shared_ext'
-  soname_spec='$libname$release$major$shared_ext'
+  if test "x$enable_debug" = xyes
+  then
+    library_names_spec='${libname}d$release$major$shared_ext ${libname}d$shared_ext'
+    soname_spec='${libname}d$release$major$shared_ext'
+  else
+    library_names_spec='$libname$release$major$shared_ext $libname$shared_ext'
+    soname_spec='$libname$release$major$shared_ext'
+  fi
   shlibpath_overrides_runpath=yes
   shlibpath_var=DYLD_LIBRARY_PATH
   shrext_cmds='`test .$module = .yes && echo .so || echo .dylib`'
@@ -15463,8 +15469,14 @@ linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
-  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
-  soname_spec='$libname$release$shared_ext$major'
+  if test "x$enable_debug" = xyes
+  then
+    library_names_spec='${libname}d$release$shared_ext$versuffix ${libname}d$release$shared_ext$major ${libname}d$shared_ext'
+    soname_spec='${libname}d$release$shared_ext$major'
+  else
+    library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+    soname_spec='$libname$release$shared_ext$major'
+  fi
   finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
