diff --git a/Modules/ThirdParty/TIFF/CMakeLists.txt b/Modules/ThirdParty/TIFF/CMakeLists.txt
index c95b6d10..b66bd704 100644
--- a/Modules/ThirdParty/TIFF/CMakeLists.txt
+++ b/Modules/ThirdParty/TIFF/CMakeLists.txt
@@ -12,16 +12,11 @@ if(ITK_USE_SYSTEM_TIFF)
   set(ITKTIFF_SYSTEM_INCLUDE_DIRS
     ${TIFF_INCLUDE_DIR}
     )
-  set(ITKTIFF_LIBRARIES "${TIFF_LIBRARIES}")
+  set(ITKTIFF_LIBRARIES "TIFF::TIFF")
   set(ITKTIFF_NO_SRC 1)
 
-  # Check availability of different methods to access TIFF's field
-  # data structure.
-  include(CheckTypeSize)
-  set(CMAKE_REQUIRED_INCLUDES ${ITKTIFF_SYSTEM_INCLUDE_DIRS})
-  set(CMAKE_EXTRA_INCLUDE_FILES "tiffio.h")
-  check_type_size(TIFFFieldReadCount ITK_TIFF_HAS_TIFFFieldReadCount)
-  check_type_size(TIFFField*  ITK_TIFF_HAS_TIFFField)
+  set(ITK_TIFF_HAS_TIFFFieldReadCount 1)
+  set(ITK_TIFF_HAS_TIFFField 1)
 else()
   set(ITKTIFF_INCLUDE_DIRS
     ${ITKTIFF_BINARY_DIR}/src
