diff --git i/Modules/Core/Common/include/itkNumberToString.h w/Modules/Core/Common/include/itkNumberToString.h
index 9d00b28..5d477b8 100644
--- i/Modules/Core/Common/include/itkNumberToString.h
+++ w/Modules/Core/Common/include/itkNumberToString.h
@@ -18,7 +18,7 @@
 #ifndef __itkNumberToString_h
 #define __itkNumberToString_h
 
-#include "double-conversion.h"
+#include <double-conversion.h>
 #include "itkMacro.h"
 #include <string>
 
