diff --git i/Modules/ThirdParty/VNL/src/vxl/v3p/netlib/linalg/lsqrBase.cxx w/Modules/ThirdParty/VNL/src/vxl/v3p/netlib/linalg/lsqrBase.cxx
index 23bbb78..09367af 100644
--- i/Modules/ThirdParty/VNL/src/vxl/v3p/netlib/linalg/lsqrBase.cxx
+++ w/Modules/ThirdParty/VNL/src/vxl/v3p/netlib/linalg/lsqrBase.cxx
@@ -619,7 +619,7 @@ Solve( unsigned int m, unsigned int n, const double * b, double * x )
     // See if it is time to print something.
     //----------------------------------------------------------------
     bool prnt = false;
-    if (nout > 0)
+    if (this->nout)
       {
       if (n     <=        40) prnt = true;
       if (this->itn   <=        10) prnt = true;
