diff --git i/CMakeLists.txt w/CMakeLists.txt
index c5b28c4..2a624a0 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -20,27 +20,6 @@ include(CMakeDependentOption)
 # use ExternalProject
 include(ExternalProject)

-if( CMAKE_HOST_WIN32 )
-
-  string( LENGTH "${CMAKE_CURRENT_SOURCE_DIR}" n )
-  if( n GREATER 50 )
-    message(
-      FATAL_ERROR
-      "ITK source code directory path length is too long (${n} > 50)."
-      "Please move the ITK source code directory to a directory with a shorter path."
-      )
-  endif()
-
-  string( LENGTH "${CMAKE_CURRENT_BINARY_DIR}" n )
-  if( n GREATER 50 )
-    message(
-      FATAL_ERROR
-      "ITK build directory path length is too long (${n} > 50)."
-      "Please set the ITK build directory to a directory with a shorter path."
-      )
-  endif()
-
-endif()

 set(CMAKE_MODULE_PATH ${ITK_SOURCE_DIR}/CMake ${CMAKE_MODULE_PATH})
 include(PreventInSourceBuilds)
