diff --git c/Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt i/Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
index aa0e098..831ad4a 100644
--- c/Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
+++ i/Modules/ThirdParty/HDF5/src/itkhdf5/CMakeLists.txt
@@ -324,7 +324,7 @@ IF (MSVC)
   SET (CMAKE_MFC_FLAG 0)
 ENDIF (MSVC)
 
-SET (MAKE_SYSTEM)
+SET (MAKE_SYSTEM 1)
 IF (CMAKE_BUILD_TOOL MATCHES "make")
   SET (MAKE_SYSTEM 1)
 ENDIF (CMAKE_BUILD_TOOL MATCHES "make")
