Index: source/Modules/ThirdParty/HDF5/src/itkhdf5/src/H5Omtime.c
===================================================================
--- source/Modules/ThirdParty/HDF5/src/itkhdf5/src/H5Omtime.c	(revision 2870)
+++ source/Modules/ThirdParty/HDF5/src/itkhdf5/src/H5Omtime.c	(revision 2871)
@@ -28,6 +28,13 @@
 #include "H5Opkg.h"             /* Object headers			*/


+#if (defined (_MSC_VER) && (_MSC_VER>=1900))
+ /* These overrides are needed for >= VC13, because config does not recognize it correctly in this version. */
+#define timezone _timezone
+#define tzname _tzname
+#define daylight _daylight
+#endif
+
 static void *H5O_mtime_new_decode(H5F_t *f, hid_t dxpl_id, H5O_t *open_oh,
     unsigned mesg_flags, unsigned *ioflags, const uint8_t *p);
 static herr_t H5O_mtime_new_encode(H5F_t *f, hbool_t disable_shared, uint8_t *p, const void *_mesg);
Index: source/Modules/ThirdParty/TIFF/src/itktiff/tif_config.h.in
===================================================================
--- source/Modules/ThirdParty/TIFF/src/itktiff/tif_config.h.in	(revision 2870)
+++ source/Modules/ThirdParty/TIFF/src/itktiff/tif_config.h.in	(revision 2871)
@@ -131,7 +131,12 @@
 #cmakedefine HAVE_PTHREAD

 /* Define to 1 if you have the <search.h> header file. */
+/* On VC13 recognition still does not work fine although it exists, thus always define it. */
+#if defined(_MSC_VER) && (_MSC_VER >= 1900)
+#define HAVE_SEARCH_H 1
+#else
 #undef HAVE_SEARCH_H
+#endif

 /* Define to 1 if you have the `setmode' function. */
 #undef HAVE_SETMODE
