diff --git c/Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h i/Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h
index e75a220..60d185e 100644
--- c/Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h
+++ i/Modules/ThirdParty/OpenJPEG/src/openjpeg/opj_includes.h
@@ -92,7 +92,7 @@ Most compilers implement their own version of this keyword ...
 
 /* MSVC 64bits doesn't support _asm */
 #if !defined(_WIN64)
-static INLINE long lrintf(float f){
+static INLINE long opj_lrintf(float f){
   int i;
 
   _asm{
@@ -103,7 +103,7 @@ static INLINE long lrintf(float f){
   return i;
 }
 #else
-static INLINE long lrintf(float x){
+static INLINE long opj_lrintf(float x){
   long r;
   if (x>=0.f)
   {
