Index: source/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h
===================================================================
--- source/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h	(revision 3030)
+++ source/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler.h	(revision 5171)
@@ -127,7 +127,13 @@
 #   define VCL_GCC_40
 #  endif
 # else
-#  error "Dunno about this gcc"
+#  if (__GNUC__ >= 5 )
+     // Just try it with setting highest known definition compiler version.
+#   define VCL_GCC_4
+#   define VCL_GCC_41
+#  else
+#    error "Dunno about this gcc"
+#  endif
 # endif
 #endif

