diff --git i/Source/WebCore/html/canvas/CanvasPattern.cpp w/Source/WebCore/html/canvas/CanvasPattern.cpp
index 77cdeb2..59b3240 100644
--- i/Source/WebCore/html/canvas/CanvasPattern.cpp
+++ w/Source/WebCore/html/canvas/CanvasPattern.cpp
@@ -20,7 +20,7 @@
  * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
  * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
 #include "config.h"
@@ -28,7 +28,7 @@
 
 #include "ExceptionCode.h"
 #include "Image.h"
-#include "Pattern.h"
+#include "../../platform/graphics/Pattern.h"
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
diff --git i/Source/WebCore/platform/graphics/qt/PatternQt.cpp w/Source/WebCore/platform/graphics/qt/PatternQt.cpp
index 8795f35..35362de 100644
--- i/Source/WebCore/platform/graphics/qt/PatternQt.cpp
+++ w/Source/WebCore/platform/graphics/qt/PatternQt.cpp
@@ -20,11 +20,11 @@
  * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
  * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
 #include "config.h"
-#include "Pattern.h"
+#include "../Pattern.h"
 
 #include "Image.h"
 
