diff --git i/Configurations/50-masm.conf w/Configurations/50-masm.conf
index 2c55ddd..a738511 100644
--- i/Configurations/50-masm.conf
+++ w/Configurations/50-masm.conf
@@ -19,5 +19,6 @@ my %targets = (
         bn_asm_src      => sub { return undef unless @_;
                                  my $r=join(" ",@_); $r=~s|asm/x86_64-gcc|bn_asm|; $r; },
         perlasm_scheme   => "masm",
+        multilib         => "-x64",
     },
 );
