diff --git c/qtbase/src/widgets/kernel/qwidget.cpp i/qtbase/src/widgets/kernel/qwidget.cpp
index db08a1d04..259b05daf 100644
--- c/qtbase/src/widgets/kernel/qwidget.cpp
+++ i/qtbase/src/widgets/kernel/qwidget.cpp
@@ -5618,10 +5618,13 @@ void QWidgetPrivate::drawWidget(QPaintDevice *pdev, const QRegion &rgn, const QP
                 } else if (!backingStore) {
                     // We are not drawing to a backingstore: fall back to QImage
                     QImage img = grabFramebuffer();
+/* MEVIS patch begin - debugging showed that it is obviously not true that grabFramebuffer always returns RGB32,
+                       and for screenshots we need to drop the alpha
                     // grabFramebuffer() always sets the format to RGB32
                     // regardless of whether it is transparent or not.
                     if (img.format() == QImage::Format_RGB32)
                         img.reinterpretAsFormat(QImage::Format_ARGB32_Premultiplied);
+MEVIS patch end */
                     QPainter p(q);
                     p.drawImage(q->rect(), img);
                     skipPaintEvent = true;
