diff --git c/qtbase/src/widgets/itemviews/qabstractitemview.cpp i/qtbase/src/widgets/itemviews/qabstractitemview.cpp
index e1e45143b..6d0516964 100644
--- c/qtbase/src/widgets/itemviews/qabstractitemview.cpp
+++ i/qtbase/src/widgets/itemviews/qabstractitemview.cpp
@@ -2668,7 +2668,10 @@ bool QAbstractItemView::edit(const QModelIndex &index, EditTrigger trigger, QEve
     if (!d->shouldEdit(trigger, d->model->buddy(index)))
         return false;
 
-    if (d->delayedEditing.isActive())
+    if (trigger == AllEditTriggers) {
+        // editing is enforced now
+        d->delayedEditing.stop();
+    } else if (d->delayedEditing.isActive())
         return false;
 
     // we will receive a mouseButtonReleaseEvent after a
