diff --git c/qtbase/src/widgets/styles/qfusionstyle.cpp i/qtbase/src/widgets/styles/qfusionstyle.cpp
index 64954dc83..8fff11d77 100644
--- c/qtbase/src/widgets/styles/qfusionstyle.cpp
+++ i/qtbase/src/widgets/styles/qfusionstyle.cpp
@@ -1993,6 +1993,14 @@ void QFusionStyle::drawComplexControl(ComplexControl control, const QStyleOption
                 frame.lineWidth = groupBox->lineWidth;
                 frame.midLineWidth = groupBox->midLineWidth;
                 frame.rect = proxy()->subControlRect(CC_GroupBox, option, SC_GroupBoxFrame, widget);
+                // --------- begin mevis -----------
+                // drawPrimitive(PE_FrameGroupBox,...) always assumes that the group box has a title and adds a top margin.
+                // We adapt the rect if the title is empty to prevent that there is an empty top margin.
+                if (groupBox->text.isEmpty() && !(groupBox->subControls & SC_GroupBoxCheckBox)) {
+                  int topMargin = qMax(pixelMetric(PM_ExclusiveIndicatorHeight), option->fontMetrics.height());
+                  frame.rect = frame.rect.adjusted(0, -topMargin, 0, 0);
+                }
+                // --------- end mevis -------------
                 proxy()->drawPrimitive(PE_FrameGroupBox, &frame, painter, widget);
             }
 
@@ -3195,8 +3203,14 @@ QSize QFusionStyle::sizeFromContents(ContentsType type, const QStyleOption *opti
         }
         break;
     case CT_GroupBox:
-        if (option) {
+        if (const QStyleOptionGroupBox *groupBox = qstyleoption_cast<const QStyleOptionGroupBox *>(option)) {
             int topMargin = qMax(pixelMetric(PM_ExclusiveIndicatorHeight), option->fontMetrics.height()) + groupBoxTopMargin;
+            // --------- begin mevis -----------
+            // if no title exists, do not add space for it to the top margin
+            if (groupBox->text.isEmpty() && !(groupBox->subControls & SC_GroupBoxCheckBox)) {
+              topMargin = groupBoxTopMargin;
+            }
+            // --------- end mevis -------------
             newSize += QSize(10, topMargin); // Add some space below the groupbox
         }
         break;
@@ -3472,6 +3486,12 @@ QRect QFusionStyle::subControlRect(ComplexControl control, const QStyleOptionCom
                 int margin = 3;
                 int leftMarginExtension = 0;
                 int topMargin = qMax(pixelMetric(PM_ExclusiveIndicatorHeight), option->fontMetrics.height()) + groupBoxTopMargin;
+                // --------- begin mevis -----------
+                // if the title is empty, then do not add space for it to the top margin
+                if (groupBox->text.isEmpty() && !(groupBox->subControls & SC_GroupBoxCheckBox)) {
+                  topMargin = groupBoxTopMargin;
+                }
+                // --------- end mevis -------------
                 return frameRect.adjusted(leftMarginExtension + margin, margin + topMargin, -margin, -margin - groupBoxBottomMargin);
             }
 
