diff --git i/qtbase/mkspecs/features/qt_functions.prf w/qtbase/mkspecs/features/qt_functions.prf
index 1903e50..47b5693 100644
--- i/qtbase/mkspecs/features/qt_functions.prf
+++ w/qtbase/mkspecs/features/qt_functions.prf
@@ -4,7 +4,7 @@ defineReplace(qtPlatformTargetSuffix) {
     CONFIG(debug, debug|release) {
         !debug_and_release|build_pass {
             mac: return($${suffix}_debug)
-            win32: return($${suffix}d)
+            win32|linux: return($${suffix}d)
         }
     }
     return($$suffix)
diff --git i/qtbase/mkspecs/features/qt_helper_lib.prf w/qtbase/mkspecs/features/qt_helper_lib.prf
index 2cb54fc..2c59d92 100644
--- i/qtbase/mkspecs/features/qt_helper_lib.prf
+++ w/qtbase/mkspecs/features/qt_helper_lib.prf
@@ -59,7 +59,7 @@ THE_TARGET = $$qt5LibraryTarget($$TARGET)
         "QMAKE_INCDIR_$${ucmodule} = $$val_escape(MODULE_INCLUDEPATH)" \
         "QMAKE_DEFINES_$${ucmodule} = $$val_escape(MODULE_DEFINES)"
     debug_and_release {
-        win32: \
+        win32|linux: \
             MODULE_DEBUG_LIBS = $$DESTDIR/$$prefix$${TARGET}d.$$suffix
         else: darwin: \
             MODULE_DEBUG_LIBS = $$DESTDIR/$$prefix$${TARGET}_debug.$$suffix
diff --git i/qtdeclarative/src/qml/qml/qqmlimport.cpp w/qtdeclarative/src/qml/qml/qqmlimport.cpp
index ba8dce4..d29e5cc 100644
--- i/qtdeclarative/src/qml/qml/qqmlimport.cpp
+++ w/qtdeclarative/src/qml/qml/qqmlimport.cpp
@@ -1926,7 +1926,12 @@ QString QQmlImportDatabase::resolvePlugin(QQmlTypeLoader *typeLoader,
     };
 # else  // Unix
     static const QString prefix = QLatin1String("lib");
-    static const QStringList suffixes = { QLatin1String(".so") };
+    static const QStringList suffixes = {
+# ifdef QT_DEBUG
+        QLatin1String("d.so"), // try a qmake-style debug build first
+# endif
+        QLatin1String(".so")
+    };
 #endif
 
     return resolvePlugin(typeLoader, qmldirPath, qmldirPluginPath, baseName, suffixes, prefix);
