diff --git a/dcmdata/libsrc/dcuid.cc b/dcmdata/libsrc/dcuid.cc
index a95eb61..1595971 100644
--- a/dcmdata/libsrc/dcuid.cc
+++ b/dcmdata/libsrc/dcuid.cc
@@ -1795,9 +1795,20 @@ static long gethostid(void)
     /* concatenate the host specific elements and compute a 32-bit checksum */
     crc.addBlock(&result /*ip address*/, OFstatic_cast(unsigned long, sizeof(result)));
     // add the MAC addresses of all adapters
+    char* replacementMac = getenv("DCMTK_REPLACEMENT_MAC");
+    if (replacementMac && strlen(replacementMac) >= 12)
+    {
+      // On virtual machines a MAC address can be defined via environment variable.
+      crc.addBlock(replacementMac, sizeof(replacementMac));
+    }
+    else
     {
       OFVector<Uint8> buffer;
-      for (IP_ADAPTER_INFO* adapterInfo = getMACAddresses(buffer); adapterInfo; adapterInfo = adapterInfo->Next)
+      IP_ADAPTER_INFO* adapterInfo = getMACAddresses(buffer);
+      if (!adapterInfo) {
+        DCMDATA_WARN("Could not read Mac Address while generating UUID. Set environment variable DCMTK_REPLACEMENT_MAC if no network adapter is installed.");
+      }
+      for (; adapterInfo; adapterInfo = adapterInfo->Next)
         crc.addBlock(adapterInfo->Address, adapterInfo->AddressLength);
     }
     crc.addBlock(&serialNumber, OFstatic_cast(unsigned long, sizeof(serialNumber)));
