diff --git a/dcmnet/apps/CMakeLists.txt b/dcmnet/apps/CMakeLists.txt
index 48061ec..db8b001 100644
--- a/dcmnet/apps/CMakeLists.txt
+++ b/dcmnet/apps/CMakeLists.txt
@@ -1,3 +1,6 @@
+# enable on the fly conversion for storescu
+ADD_DEFINITIONS("-DON_THE_FLY_COMPRESSION")
+
 # declare additional include directories needed for compression support
 include_directories("${dcmjpls_SOURCE_DIR}/include" "${dcmjpeg_SOURCE_DIR}/include" "${dcmimage_SOURCE_DIR}/include" "${dcmimgle_SOURCE_DIR}/include")
 
@@ -34,7 +37,7 @@ foreach(PROGRAM dcmrecv echoscu findscu storescp storescu)
 endforeach()
 
 # uncomment the following line if "storescu" is compiled with ON_THE_FLY_COMPRESSION defined
-#DCMTK_TARGET_LINK_MODULES(storescu dcmjpls dcmjpeg dcmimage)
+DCMTK_TARGET_LINK_MODULES(storescu dcmjpls dcmjpeg dcmimage)
 
 # "dcmsend" always needs compression support
 DCMTK_TARGET_LINK_MODULES(dcmsend dcmjpls dcmjpeg dcmimage)
