diff --git a/CMake/dcmtkPrepare.cmake b/CMake/dcmtkPrepare.cmake
index 5800b96..0de56dd 100644
--- a/CMake/dcmtkPrepare.cmake
+++ b/CMake/dcmtkPrepare.cmake
@@ -720,7 +720,7 @@ function(DCMTK_TEST_SOCKET_LIBRARY NAME SYMBOL)
   endif()
 endfunction()
 
-DCMTK_TEST_SOCKET_LIBRARY(nsl "gethostbyname")
+#DCMTK_TEST_SOCKET_LIBRARY(nsl "gethostbyname")
 DCMTK_TEST_SOCKET_LIBRARY(socket "socket")
 
 #-----------------------------------------------------------------------------
