diff --git a/CMake/dcmtkPrepare.cmake b/CMake/dcmtkPrepare.cmake
index 365666d..eea3651 100644
--- a/CMake/dcmtkPrepare.cmake
+++ b/CMake/dcmtkPrepare.cmake
@@ -764,7 +764,7 @@ function(DCMTK_TEST_SOCKET_LIBRARY NAME SYMBOL)
   endif()
 endfunction()
 
-DCMTK_TEST_SOCKET_LIBRARY(nsl "gethostbyname")
+#DCMTK_TEST_SOCKET_LIBRARY(nsl "gethostbyname")
 DCMTK_TEST_SOCKET_LIBRARY(socket "socket")
 
 #-----------------------------------------------------------------------------
