diff --git i/dcmdata/libsrc/cmdlnarg.cc w/dcmdata/libsrc/cmdlnarg.cc
index ad5afe1..706ae91 100644
--- i/dcmdata/libsrc/cmdlnarg.cc
+++ w/dcmdata/libsrc/cmdlnarg.cc
@@ -112,7 +112,22 @@ void prepareCmdLineArgs(int& /* argc */, char** /* argv */,
     if (fileno(stderr) != fileno(stdout))
     {
         /* duplicate the stderr file descriptor to be the same as stdout */
-        close(fileno(stderr));
+    const int fd_stderr = fileno( stderr );
+    if ( fd_stderr < 0 )
+    {
+      // stderr has no valid file descriptor because
+      // it may already have been mapped by the parent process
+      return;
+    }
+
+    const int ret = close( fd_stderr );
+    if ( ret == -1 )
+    {
+      // in case we cannot close stderr we stop here
+      // since the following code will fail
+      return;
+    }
+
         int fderr = dup(fileno(stdout));
         if (fderr != fileno(stderr))
         {
