diff --git c/MeVisRecreateBuiltinDicomDict.bat i/MeVisRecreateBuiltinDicomDict.bat
new file mode 100644
index 0000000..8ae7248
--- /dev/null
+++ i/MeVisRecreateBuiltinDicomDict.bat
@@ -0,0 +1,15 @@
+rem This is an small script to re-create the DCMTK dicom dictionary
+rem it requires building of DCMTK first, afterwards the paths need to be adapted and it can be run
+rem and creates the new dcdictbi.cc.
+rem We do this because we want the (old/retired) acr nema tags to be part of DCMTK for backwards campatibility...
+rem This has to be done again when we upgrade DCMTK.
+
+set BASE=D:\repository\ThirdParty\Sources\fileformat\dcmtk
+set PATH=d:\repository\Thirdparty\lib;%PATH%
+
+rem join the two dicts 
+copy %BASE%\dcmdata\data\dicom.dic %BASE%\dcmdata\data\both.dic
+type %BASE%\dcmdata\data\acrnema.dic >> %BASE%\dcmdata\data\both.dic
+
+rem create the dictionary
+%BASE%\bin\Debug\mkdictbi.exe %BASE%\dcmdata\data\both.dic %BASE%\dcmdata\data\private.dic >%BASE%\dcmdata\libsrc\dcdictbi.cc
