diff --git i/CMake/osconfig.h.in w/CMake/osconfig.h.in
index c3cf55c..4454d57 100644
--- i/CMake/osconfig.h.in
+++ w/CMake/osconfig.h.in
@@ -1075,7 +1075,11 @@ typedef unsigned long ulong;
 
 #if _MSC_VER >= 1400                   /* Additional settings for Visual Studio 2005 and newer */
     #pragma warning( disable : 4996 )  /* disable warnings about "deprecated" C runtime functions */
-    #pragma warning( disable : 4351 )  /* disable warnings about "new behavior" when initializing the elements of an array */
+
+    #if _MSC_VER < 1900 /* Only available before Visual Studio 2015 */
+    #pragma warning( disable : 4351 )  // disable warnings about "new behavior" when initializing the elements of an array
+    #endif /* _MSC_VER < 1900 */
+
 #endif /* _MSC_VER >= 1400 */
 #endif /* _MSC_VER */
 
