diff --git i/dcmpstat/include/dcmtk/dcmpstat/dvpscf.h w/dcmpstat/include/dcmtk/dcmpstat/dvpscf.h
index 028fe0d..e7fbb36 100644
--- i/dcmpstat/include/dcmtk/dcmpstat/dvpscf.h
+++ w/dcmpstat/include/dcmtk/dcmpstat/dvpscf.h
@@ -741,6 +741,16 @@ class DCMTK_DCMPSTAT_EXPORT DVConfiguration
    */
   OFBool getSpoolerDeletePrintJobs();
 
+  // jj20030225
+  // want to handle configuration files separately from the rest
+  // since each print scu has its own configuration, which shall be deleted
+  // except for one
+  OFBool getSpoolerDeleteConfiguration();
+
+  // jj20030225
+  // want to handle intermediate printJob files separately from the rest
+  OFBool getSpoolerDeleteIntermediateFiles();
+
   /** returns the ALWAYSDELETETERMINATEJOBS entry for the print spooler
    *  from the section GENERAL/PRINT in the config file.
    *  @return entry if present in the config file, OFFalse otherwise.
diff --git i/dcmpstat/libsrc/dvpscf.cc w/dcmpstat/libsrc/dvpscf.cc
index 18e4359..54fa148 100644
--- i/dcmpstat/libsrc/dvpscf.cc
+++ w/dcmpstat/libsrc/dvpscf.cc
@@ -59,6 +59,9 @@ extern "C" int strncasecmp(const char *s1, const char *s2, size_t n);
 #define L0_DEFAULTILLUMINATION          "DEFAULTILLUMINATION"
 #define L0_DEFAULTREFLECTION            "DEFAULTREFLECTION"
 #define L0_DELETEPRINTJOBS              "DELETEPRINTJOBS"
+// jj20030307
+#define L0_DELETECONFIGURATIONS         "DELETECONFIGURATIONS"
+#define L0_DELETEINTERMEDIATEFILES      "DELETEINTERMEDIATEFILES"
 #define L0_DESCRIPTION                  "DESCRIPTION"
 #define L0_DETAILEDLOG                  "DETAILEDLOG"
 #define L0_DICOMNAME                    "DICOMNAME"
@@ -579,6 +582,22 @@ OFBool DVConfiguration::getSpoolerDeletePrintJobs()
   return getConfigBoolEntry(L2_GENERAL, L1_PRINT, L0_DELETEPRINTJOBS, OFFalse);
 }
 
+// jj20030225
+// want to handle configuration files separately from the rest
+// since each print scu has its own configuration, which shall be deleted
+// except for one
+OFBool DVConfiguration::getSpoolerDeleteConfiguration()
+{ 
+  return getConfigBoolEntry( L2_GENERAL, L1_PRINT, L0_DELETECONFIGURATIONS, OFFalse );
+}
+
+// jj20030307 Bug# 1656
+// want to handle intermediate printJob files separately from the rest
+OFBool DVConfiguration::getSpoolerDeleteIntermediateFiles()
+{
+  return getConfigBoolEntry( L2_GENERAL, L1_PRINT, L0_DELETEINTERMEDIATEFILES, OFFalse );
+}
+
 OFBool DVConfiguration::getSpoolerAlwaysDeleteTerminateJobs()
 {
   return getConfigBoolEntry(L2_GENERAL, L1_PRINT, L0_ALWAYSDELETETERMINATEJOBS, OFFalse);
