diff --git i/dcmpstat/apps/dcmpsprt.cc w/dcmpstat/apps/dcmpsprt.cc
index 574f760..4d1fa10 100644
--- i/dcmpstat/apps/dcmpsprt.cc
+++ w/dcmpstat/apps/dcmpsprt.cc
@@ -136,6 +136,10 @@ int main(int argc, char *argv[])
     int                       opt_LUTshape = 0; // 0=use SCP default, 1=IDENTITY, 2=LIN OD.
     OFBool                    opt_inverse_plut = OFFalse;
     OFBool                    opt_spool = OFFalse;
+
+    // jj20040701: make logging more visible
+    const char*               opt_jobPrefix = NULL;    /* prefix for print client job and log files */
+    
     const char *              opt_mediumtype = NULL;
     const char *              opt_destination = NULL;
     const char *              opt_sessionlabel = NULL;
@@ -217,6 +221,9 @@ int main(int argc, char *argv[])
      cmd.addOption("--owner",                    1, "[v]alue: string",
                                                     "set film session owner ID to v");
 
+    // jj20040701: make logging more visible
+    cmd.addOption( "--jobPrefix", "[p]refix: string (prefix for print client job and log file" );
+
     cmd.addGroup("annotation options:");
      cmd.addOption("--no-annotation",               "do not create annotation (default)");
      cmd.addOption("--annotation",        "-a",  1, "[t]ext: string",
@@ -368,6 +375,12 @@ int main(int argc, char *argv[])
       if (cmd.findOption("--img-smoothing"))     app.checkValue(cmd.getValue(opt_img_smoothing));
       if (cmd.findOption("--img-configinfo"))    app.checkValue(cmd.getValue(opt_img_configuration));
 
+      // jj20040701: make logging more visible
+      if( cmd.findOption("--jobPrefix") ) {
+        app.checkConflict( "--jobPrefix", "--nospool", (!opt_spool) );
+        app.checkValue( cmd.getValue(opt_jobPrefix) );
+      }
+
       /* film session options */
       if (cmd.findOption("--medium-type"))
       {
@@ -418,7 +431,10 @@ int main(int argc, char *argv[])
       const char *imageFile = NULL;
       const char *pstateFile = NULL;
       int paramCount = cmd.getParamCount();
-      for (int param = 1; param <= paramCount; param++)
+
+      // jj20040701: if there is a jobPrefix defined, ignore this string
+      int start = opt_jobPrefix ? 2 : 1;
+      for (int param = start; param <= paramCount; param++)
       {
         cmd.getParam(param, imageFile);
         pstateFile = NULL;
@@ -440,7 +456,9 @@ int main(int argc, char *argv[])
         return 10;
       }
     }
-    DVInterface dvi(opt_cfgName);
+
+    // jj20040702: use print job prefix for printJobIdentifier
+    DVInterface dvi(opt_cfgName, OFFalse, opt_jobPrefix);
 
     if (opt_printerID && (EC_Normal != dvi.setCurrentPrinter(opt_printerID)))
       OFLOG_WARN(dcmpsprtLogger, "unable to select printer '" << opt_printerID << "', ignoring.");
@@ -601,6 +619,17 @@ int main(int argc, char *argv[])
           OFLOG_FATAL(dcmpsprtLogger, "out of memory error: cannot allocate print bitmap");
           return 10;
         }
+        
+        // jj20050526: delete loaded files
+        if( currentPState ) {
+          OFLOG_INFO(dcmpsprtLogger, "deleting image file '" << currentImage << "' with presentation state '" << currentPState << "'");
+          unlink( currentPState );
+          unlink( currentImage );
+        }
+        else {
+          OFLOG_INFO(dcmpsprtLogger, "deleting image file '" << currentImage);
+          unlink( currentImage );
+        }
       } else {
         OFLOG_FATAL(dcmpsprtLogger, "internal error - odd number of filenames");
         return 10;
