diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c36a84..e0f23ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,9 @@ cmake_minimum_required(VERSION 3.12)
 
 PROJECT(vigra)
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 # read the current version from config_version.hxx
 file(READ ${vigra_SOURCE_DIR}/include/vigra/config_version.hxx VIGRA_VERSION_FILE)
 string(REGEX MATCH "VIGRA_VERSION_MAJOR[ \t\n]+[^ \t\n]+" VIGRA_VERSION_MAJOR ${VIGRA_VERSION_FILE})
diff --git a/src/impex/CMakeLists.txt b/src/impex/CMakeLists.txt
index 98a257a..93b226a 100644
--- a/src/impex/CMakeLists.txt
+++ b/src/impex/CMakeLists.txt
@@ -68,11 +68,11 @@ set(SOVERSION 11)  # increment this after changing the vigraimpex library
 IF(MACOSX)
     SET_TARGET_PROPERTIES(vigraimpex PROPERTIES VERSION ${SOVERSION}.${vigra_version}
                           SOVERSION ${SOVERSION} INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}"
-                          CXX_STANDARD 11)
+                          CXX_STANDARD 17)
 ELSE()
     SET_TARGET_PROPERTIES(vigraimpex PROPERTIES VERSION ${SOVERSION}.${vigra_version}
                           SOVERSION ${SOVERSION}
-                          CXX_STANDARD 11)
+                          CXX_STANDARD 17)
 ENDIF()
 
 IF(JPEG_FOUND)
