diff --git a/include/vigra/basicimage.hxx b/include/vigra/basicimage.hxx
index 84a05cc..11b716e 100644
--- a/include/vigra/basicimage.hxx
+++ b/include/vigra/basicimage.hxx
@@ -582,7 +582,7 @@ class BasicImage
     typedef Alloc allocator_type;
 
     typedef Alloc Allocator;
-    typedef typename Alloc::template rebind<PIXELTYPE *>::other LineAllocator;
+    typedef typename std::allocator_traits<Alloc>::template rebind_alloc<PIXELTYPE *> LineAllocator;
 
         /** construct image of size 0x0
         */
