diff --git a/src/qhttpserver.cpp b/src/qhttpserver.cpp
index 07e4a85..fb7d075 100644
--- a/src/qhttpserver.cpp
+++ b/src/qhttpserver.cpp
@@ -106,10 +106,17 @@ void QHttpServer::newConnection()
     }
 }
 
-bool QHttpServer::listen(const QHostAddress &address, quint16 port)
+void QHttpServer::setTcpServer(QTcpServer* server)
 {
     Q_ASSERT(!m_tcpServer);
-    m_tcpServer = new QTcpServer(this);
+    m_tcpServer = server;
+}
+
+bool QHttpServer::listen(const QHostAddress &address, quint16 port)
+{
+    if (!m_tcpServer) {
+        m_tcpServer = new QTcpServer;
+    }
 
     bool couldBindToPort = m_tcpServer->listen(address, port);
     if (couldBindToPort) {
diff --git a/src/qhttpserver.h b/src/qhttpserver.h
index 46c5b4b..76d2b3e 100644
--- a/src/qhttpserver.h
+++ b/src/qhttpserver.h
@@ -81,6 +81,11 @@ public:
 
     /// Stop the server and listening for new connections.
     void close();
+
+    //! Set the QTcpServer (or a derived class) to be used. Call this BEFORE calling listen()
+    //! if you want your own QTcpServer.
+    void setTcpServer(QTcpServer* server);
+
 Q_SIGNALS:
     /// Emitted when a client makes a new request to the server.
     /** The slot should use the given @c request and @c response
