diff --git i/tnt_sparse_matrix_csr.h w/tnt_sparse_matrix_csr.h
index 0d4fde1..f209f5f 100644
--- i/tnt_sparse_matrix_csr.h
+++ w/tnt_sparse_matrix_csr.h
@@ -59,7 +59,7 @@ public:
 
 	Sparse_Matrix_CompRow(const Sparse_Matrix_CompRow &S);
 	Sparse_Matrix_CompRow(int M, int N, int nz, const T *val, 
-						const int *r, const int *c);
+						int *r, int *c);
     
 
 
@@ -93,7 +93,7 @@ public:
 */
 template <class T>
 Sparse_Matrix_CompRow<T>::Sparse_Matrix_CompRow(int M, int N, int nz,
-	const T *val, const int *r, const int *c) : val_(nz,val), 
+	const T *val, int *r, int *c) : val_(nz,val), 
 		rowptr_(M, r), colind_(nz, c), dim1_(M), dim2_(N) {}
 
 
