diff --git a/png.c b/png.c
index cbd54e1..de9b02d 100644
--- a/png.c
+++ b/png.c
@@ -2468,8 +2468,10 @@ png_compare_ICC_profile_with_sRGB(png_const_structrp png_ptr,
                       * discourage their use, skip the 'have_md5' warning below,
                       * which is made irrelevant by this error.
                       */
-                     png_chunk_report(png_ptr, "known incorrect sRGB profile",
-                         PNG_CHUNK_ERROR);
+                     // MeVis: disabled because many PNGs have these incorrect profiles
+                     // and Qt does not make use of them.
+                     //png_chunk_report(png_ptr, "known incorrect sRGB profile",
+                     //   PNG_CHUNK_ERROR);
                   }
 
                   /* Warn that this being done; this isn't even an error since
