diff --git c/CMakeLists.txt i/CMakeLists.txt
index 6451fcf..bfb4db2 100644
--- c/CMakeLists.txt
+++ i/CMakeLists.txt
@@ -39,7 +39,9 @@ include(GNUInstallDirs)
 option(PNG_BUILD_ZLIB "Custom zlib Location, else find_package is used" OFF)

 if(NOT PNG_BUILD_ZLIB)
-  find_package(ZLIB REQUIRED)
+  set(ZLIB_FOUND 1)
+  set(ZLIB_INCLUDE_DIR ${CONAN_INCLUDE_DIRS_ZLIB})
+  set(ZLIB_LIBRARY ${CONAN_LIBS_ZLIB})
   include_directories(${ZLIB_INCLUDE_DIR})
 endif()

