diff --git i/CMakeLists.txt w/CMakeLists.txt
index e355e4d..2ee0217 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -78,7 +78,12 @@ option(PNG_HARDWARE_OPTIMIZATIONS "Enable hardware optimizations" ON)
 # Useful if zlib is being built alongside this as a sub-project.
 option(PNG_BUILD_ZLIB "Custom zlib location, else find_package is used" OFF)
 if(NOT PNG_BUILD_ZLIB)
-  find_package(ZLIB REQUIRED)
+  set(ZLIB_FOUND 1)
+  set(ZLIB_INCLUDE_DIR ${CONAN_INCLUDE_DIRS_ZLIB})
+  set(ZLIB_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_ZLIB})
+  set(ZLIB_LIBRARY ${CONAN_LIBS_ZLIB})
+  set(ZLIB_LIBRARIES ${CONAN_LIBS_ZLIB})
+
   include_directories(${ZLIB_INCLUDE_DIRS})
 endif()
 
