diff --git a/src/widgets/itemviews/qabstractitemview.cpp b/src/widgets/itemviews/qabstractitemview.cpp
index 3fd71427..5a9fadcd 100644
--- a/src/widgets/itemviews/qabstractitemview.cpp
+++ b/src/widgets/itemviews/qabstractitemview.cpp
@@ -2789,7 +2789,10 @@ bool QAbstractItemView::edit(const QModelIndex &index, EditTrigger trigger, QEve
     if (!d->shouldEdit(trigger, d->model->buddy(safeIndex)))
         return false;
 
-    if (d->delayedEditing.isActive())
+    if (trigger == AllEditTriggers) {
+        // editing is enforced now
+        d->delayedEditing.stop();
+    } else if (d->delayedEditing.isActive())
         return false;
 
     // we will receive a mouseButtonReleaseEvent after a
