diff --git a/src/corelib/kernel/qobject.cpp b/src/corelib/kernel/qobject.cpp
index 7a96918f..4d324a38 100644
--- a/src/corelib/kernel/qobject.cpp
+++ b/src/corelib/kernel/qobject.cpp
@@ -655,6 +655,58 @@ void QMetaCallEvent::placeMetaCall(QObject *object)
     }
 }
 
+// MEVIS BEGIN
+QEvent* mevis_cloneAndClearQMetaCallEvent(QEvent* event)
+{
+    if (event->type() == QEvent::MetaCall) {
+        return static_cast<QMetaCallEvent*>(event)->cloneAndClear();
+    } else {
+        return nullptr;
+    }
+}
+
+QEvent* QMetaCallEvent::cloneAndClear()
+{
+    QMetaCallEvent* newEvent = nullptr;
+    if (d.nargs_) {
+        // non-blocking call, arguments need to be copied
+        if (d.slotObj_) {
+            auto slotObjPtr = d.slotObj_.release();
+            newEvent = new QMetaCallEvent(slotObjPtr,
+                                          sender(), signalId(),
+                                          d.nargs_);
+        } else {
+            newEvent = new QMetaCallEvent(d.method_offset_, d.method_relative_, d.callFunction_,
+                                          sender(), signalId(),
+                                          d.nargs_);
+        }
+        QMetaType *sourceTypeIDs = types();
+        QMetaType *targetTypeIDs = newEvent->types();
+        void **targetArgs = newEvent->args();
+        // move args:
+        for (int i = 0; i < d.nargs_; ++i) {
+            targetTypeIDs[i] = sourceTypeIDs[i]; sourceTypeIDs[i] = {};
+            targetArgs[i] = d.args_[i]; d.args_[i] = nullptr;
+        }
+    } else {
+        // blocking call, arguments stay with sender
+        if (d.slotObj_) {
+            auto slotObjPtr = d.slotObj_.release();
+            newEvent = new QMetaCallEvent(slotObjPtr,
+                                          sender(), signalId(),
+                                          d.args_, semaphore_);
+        } else {
+            newEvent = new QMetaCallEvent(d.method_offset_, d.method_relative_, d.callFunction_,
+                                          sender(), signalId(),
+                                          d.args_, semaphore_);
+        }
+        // semaphore is moved to new event:
+        semaphore_ = nullptr;
+    }
+    return newEvent;
+}
+// MEVIS END
+
 QMetaCallEvent* QMetaCallEvent::create_impl(QtPrivate::SlotObjUniquePtr slotObj,
                                             const QObject *sender, int signal_index,
                                             size_t argc, const void* const argp[],
diff --git a/src/corelib/kernel/qobject.h b/src/corelib/kernel/qobject.h
index f6a3ebc5..7fcfd932 100644
--- a/src/corelib/kernel/qobject.h
+++ b/src/corelib/kernel/qobject.h
@@ -51,6 +51,12 @@ Q_CORE_EXPORT void qt_qFindChildren_helper(const QObject *parent, const QRegular
                                            const QMetaObject &mo, QList<void *> *list, Qt::FindChildOptions options);
 Q_CORE_EXPORT QObject *qt_qFindChild_helper(const QObject *parent, const QString &name, const QMetaObject &mo, Qt::FindChildOptions options);
 
+// MEVIS BEGIN
+// Method added for MeVisLab to allow to clone and clear a QMetaCallEvent, which is private in Qt
+#define MEVIS_QMETACALLEVENT_HACK
+Q_CORE_EXPORT QEvent* mevis_cloneAndClearQMetaCallEvent(QEvent* event);
+// MEVIS END
+
 class Q_CORE_EXPORT QObjectData
 {
     Q_DISABLE_COPY(QObjectData)
diff --git a/src/corelib/kernel/qobject_p.h b/src/corelib/kernel/qobject_p.h
index 41f1c2f8..2ab68070 100644
--- a/src/corelib/kernel/qobject_p.h
+++ b/src/corelib/kernel/qobject_p.h
@@ -359,6 +359,7 @@ public:
 private:
     int signalId_;
     const QObject *sender_;
+protected:  // added by MeVis to allow access from QMetaCallEvent    
 #if QT_CONFIG(thread)
     QSemaphore *semaphore_;
 #endif
@@ -420,6 +421,12 @@ public:
 
     virtual void placeMetaCall(QObject *object) override;
 
+    // MEVIS BEGIN
+    //! Returns a cloned QMetaCallEvent and clears the data of the original event.
+    //! Added for MeVis use case, used in mevis_cloneAndClearQMetaCallEvent
+    QEvent* cloneAndClear();
+    // MEVIS END
+
 private:
     static QMetaCallEvent *create_impl(QtPrivate::QSlotObjectBase *slotObj, const QObject *sender,
                                        int signal_index, size_t argc, const void * const argp[],
