diff --git a/src/widgets/kernel/qwidget.cpp b/src/widgets/kernel/qwidget.cpp
index 854e4da5..b2c762f0 100644
--- a/src/widgets/kernel/qwidget.cpp
+++ b/src/widgets/kernel/qwidget.cpp
@@ -7612,9 +7612,10 @@ bool QWidget::restoreGeometry(const QByteArray &geometry)
     // high DPI scaling or different levels of DPI awareness.
     if (restoredScreenWidth) {
         const qreal factor = qreal(restoredScreenWidth) / screenWidthF;
-        if (factor < 0.8 || factor > 1.25)
-            return false;
-    } else {
+        if (!maximized && !fullScreen)
+            if (factor < 0.8 || factor > 1.25)
+                return false;
+        } else {
         // Saved by Qt 5.3 and earlier, try to prevent too large windows
         // unless the size will be adapted by maximized or fullscreen.
         if (!maximized && !fullScreen && qreal(restoredFrameGeometry.width()) / screenWidthF > 1.5)
