diff --git a/cmake/QtTargetHelpers.cmake b/cmake/QtTargetHelpers.cmake
index 43034efe..dcd17180 100644
--- a/cmake/QtTargetHelpers.cmake
+++ b/cmake/QtTargetHelpers.cmake
@@ -559,7 +559,7 @@ function(qt_internal_setup_cmake_config_postfix)
                 set(default_cmake_debug_postfix "d")
             endif()
         else()
-            set(default_cmake_debug_postfix "d")
+            set(default_cmake_debug_postfix "_d")
         endif()
     elseif(APPLE)
         # Only add a suffix for explicit no-framework builds.
