diff --git a/src/plugins/platforms/windows/qwindowsdrag.cpp b/src/plugins/platforms/windows/qwindowsdrag.cpp
index b797217c..7a74beec 100644
--- a/src/plugins/platforms/windows/qwindowsdrag.cpp
+++ b/src/plugins/platforms/windows/qwindowsdrag.cpp
@@ -270,11 +270,15 @@ void QWindowsOleDropSource::createCursors()
         hotSpotScaleFactor = QHighDpiScaling::factor(platformScreen);
         pixmapScaleFactor = hotSpotScaleFactor / pixmap.devicePixelRatio();
     }
-    QPixmap scaledPixmap = qFuzzyCompare(pixmapScaleFactor, 1.0)
+    QPixmap scaledPixmap;
+    if ( hasPixmap )
+    {
+      scaledPixmap = qFuzzyCompare(pixmapScaleFactor, 1.0)
         ? pixmap
-        :  pixmap.scaled((QSizeF(pixmap.size()) * pixmapScaleFactor).toSize(),
-                         Qt::KeepAspectRatio, Qt::SmoothTransformation);
-    scaledPixmap.setDevicePixelRatio(1);
+        : pixmap.scaled((QSizeF(pixmap.size()) * pixmapScaleFactor).toSize(),
+                        Qt::KeepAspectRatio, Qt::SmoothTransformation);
+      scaledPixmap.setDevicePixelRatio(1);
+    }
 
     Qt::DropAction actions[] = { Qt::MoveAction, Qt::CopyAction, Qt::LinkAction, Qt::IgnoreAction };
     int actionCount = int(sizeof(actions) / sizeof(actions[0]));
