diff --git i/src/x86/ffiw64.c w/src/x86/ffiw64.c
index b68f69c..3808413 100644
--- i/src/x86/ffiw64.c
+++ w/src/x86/ffiw64.c
@@ -107,6 +107,10 @@ EFI64(ffi_prep_cif_machdep)(ffi_cif *cif)
   return FFI_OK;
 }
 
+#if defined(_MSC_VER)
+#pragma runtime_checks("s", off)
+#endif
+
 static void
 ffi_call_int (ffi_cif *cif, void (*fn)(void), void *rvalue,
 	      void **avalue, void *closure)
@@ -172,6 +176,10 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *rvalue,
   ffi_call_win64 (stack, frame, closure);
 }
 
+#if defined(_MSC_VER)
+#pragma runtime_checks("s", restore)
+#endif
+
 void
 EFI64(ffi_call)(ffi_cif *cif, void (*fn)(void), void *rvalue, void **avalue)
 {
