diff --git a/CMakeLists.txt b/CMakeLists.txt
index f810d5a..3f3aaa8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,11 @@ find_package(
 # SKBUILD_SOABI seems to be
 if (DEFINED SKBUILD_SOABI AND NOT "${SKBUILD_SOABI}" STREQUAL "${Python_SOABI}")
   message(WARNING "SKBUILD_SOABI=${SKBUILD_SOABI} != Python_SOABI=${Python_SOABI}; likely cross-compiling, using SOABI=${SKBUILD_SOABI} from scikit-build")
-  set(Python_SOABI "${SKBUILD_SOABI}")
+  # the setting is not correct in debug mode, so we use Python_SOABI
+  # and add the debug suffix to the prefix later
+  if (NOT MSVC)
+    set(Python_SOABI "${SKBUILD_SOABI}")
+  endif()
 endif()
 
 # legacy pyzmq env options, no PYZMQ_ prefix
@@ -166,7 +170,8 @@ elseif (NOT ZMQ_PREFIX STREQUAL "bundled")
   )
   find_library(
     LIBZMQ_LIBRARY
-    NAMES zmq
+    # workaround for lib name under Windows
+    NAMES zmq_d zmq zmq-v143-mt-gd-4_3_5 zmq-v143-mt-4_3_5
     PATHS "${ZMQ_PREFIX}/lib"
     NO_DEFAULT_PATH
   )
@@ -367,7 +372,11 @@ if (NOT TARGET "${libzmq_target}" AND LIBZMQ_LIBRARY AND LIBZMQ_INCLUDE_DIR)
   # construct target from find_library results
   # what if it was static?
   add_library(libzmq SHARED IMPORTED)
-  set_property(TARGET libzmq PROPERTY IMPORTED_LOCATION ${LIBZMQ_LIBRARY})
+  if (MSVC)
+    set_property(TARGET libzmq PROPERTY IMPORTED_IMPLIB ${LIBZMQ_LIBRARY})
+  else()
+    set_property(TARGET libzmq PROPERTY IMPORTED_LOCATION ${LIBZMQ_LIBRARY})
+  endif()
   set_property(TARGET libzmq PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${LIBZMQ_INCLUDE_DIR})
 endif()
 
@@ -431,6 +440,10 @@ endif()
 
 file(MAKE_DIRECTORY ${ZMQ_BACKEND_DEST})
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" AND MSVC)
+  set(ZMQ_EXT_NAME "${ZMQ_EXT_NAME}_d")
+endif()
+
 python_add_library(
   ${ZMQ_EXT_NAME} MODULE
   WITH_SOABI
