diff --git i/setup.py w/setup.py
index b634d94..4129b31 100755
--- i/setup.py
+++ w/setup.py
@@ -260,7 +260,7 @@ def settings_from_prefix(prefix=None):
                     break
             else:
                 warn(f"zmq.h not found in {prefix} or {prefix}/include")
-            for library_dir in [pjoin(prefix, 'lib'), prefix]:
+            for library_dir in [pjoin(prefix, 'lib'), pjoin(prefix, 'bin'), prefix]:
                 matches = glob(pjoin(library_dir, f"{libzmq_name}*.dll"))
                 if matches:
                     libzmq_path = matches[0]
