diff --git i/openvdb/openvdb/tree/Tree.h w/openvdb/openvdb/tree/Tree.h
index 609be1e..3cada0e 100644
--- i/openvdb/openvdb/tree/Tree.h
+++ w/openvdb/openvdb/tree/Tree.h
@@ -1805,9 +1805,7 @@ template<typename RootNodeType>
 inline const Name&
 Tree<RootNodeType>::treeType()
 {
-    static std::once_flag once;
-    std::call_once(once, []()
-    {
+    static bool once = []() {
         std::vector<Index> dims;
         Tree::getNodeLog2Dims(dims);
         std::ostringstream ostr;
@@ -1816,7 +1814,9 @@ Tree<RootNodeType>::treeType()
             ostr << "_" << dims[i];
         }
         sTreeTypeName.reset(new Name(ostr.str()));
-    });
+        return true;
+    }();
+
     return *sTreeTypeName;
 }
 
