diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd8f515..c77f8f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -588,12 +588,12 @@ ENDIF()
 
 IF( NOT IOS )
   IF( NOT ASSIMP_BUILD_MINIZIP )
-    use_pkgconfig(UNZIP minizip)
+    #use_pkgconfig(UNZIP minizip)
   ENDIF()
 ELSE ()
   IF( NOT BUILD_SHARED_LIBS )
     IF( NOT ASSIMP_BUILD_MINIZIP )
-      USE_PKGCONFIG(UNZIP minizip)
+      #USE_PKGCONFIG(UNZIP minizip)
     ENDIF()
   ENDIF ()
 ENDIF ()
