diff --git a/CMakeLists.txt b/CMakeLists.txt
index f1e2d2d..d01a0fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -526,12 +526,12 @@ ENDIF()
 
 IF( NOT IOS )
   IF( NOT ASSIMP_BUILD_MINIZIP )
-    use_pkgconfig(UNZIP minizip)
+    #use_pkgconfig(UNZIP minizip)
   ENDIF()
 ELSE ()
   IF( NOT BUILD_SHARED_LIBS )
     IF( NOT ASSIMP_BUILD_MINIZIP )
-      use_pkgconfig(UNZIP minizip)
+      #use_pkgconfig(UNZIP minizip)
     ENDIF()
   ENDIF ()
 ENDIF ()
