diff --git c/CMakeLists.txt i/CMakeLists.txt
index 405f0b4..2516513 100644
--- c/CMakeLists.txt
+++ i/CMakeLists.txt
@@ -497,12 +497,12 @@ ENDIF()
 
 IF( NOT IOS )
   IF( NOT ASSIMP_BUILD_MINIZIP )
-    use_pkgconfig(UNZIP minizip)
+    #use_pkgconfig(UNZIP minizip)
   ENDIF()
 ELSE ()
   IF( NOT BUILD_SHARED_LIBS )
     IF( NOT ASSIMP_BUILD_MINIZIP )
-      use_pkgconfig(UNZIP minizip)
+      #use_pkgconfig(UNZIP minizip)
     ENDIF()
   ENDIF ()
 ENDIF ()
