diff --git i/CMakeLists.txt w/CMakeLists.txt
index 570c293..a687dc7 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -499,12 +499,12 @@ ENDIF()
 
 IF( NOT IOS )
   IF( NOT ASSIMP_BUILD_MINIZIP )
-    use_pkgconfig(UNZIP minizip)
+    #use_pkgconfig(UNZIP minizip)
   ENDIF()
 ELSE ()
   IF( NOT BUILD_SHARED_LIBS )
     IF( NOT ASSIMP_BUILD_MINIZIP )
-      use_pkgconfig(UNZIP minizip)
+      #use_pkgconfig(UNZIP minizip)
     ENDIF()
   ENDIF ()
 ENDIF ()
