diff --git i/cmake-modules/FindZLIB.cmake w/cmake-modules/FindZLIB.cmake
index 95eb515..28f694f 100644
--- i/cmake-modules/FindZLIB.cmake
+++ w/cmake-modules/FindZLIB.cmake
@@ -31,7 +31,7 @@ clear_if_changed(ZLIB_PREFIX_PATH
   ZLIB_INCLUDE_DIR
 )
 
-set(ZLIB_LIBRARY_NAMES z zlib zdll)
+set(ZLIB_LIBRARY_NAMES z zlib zdll zlibstatic)
 get_debug_names(ZLIB_LIBRARY_NAMES)
 
 use_pkgconfig(ZLIB_PKGC zzip-zlib-config)
