diff --git a/Lib/multiprocessing/pool.py b/Lib/multiprocessing/pool.py
index f979890..10f8a29 100644
--- a/Lib/multiprocessing/pool.py
+++ b/Lib/multiprocessing/pool.py
@@ -17,6 +17,7 @@ import collections
 import itertools
 import os
 import queue
+import sys
 import threading
 import time
 import traceback
@@ -206,6 +207,9 @@ class Pool(object):
         if maxtasksperchild is not None:
             if not isinstance(maxtasksperchild, int) or maxtasksperchild <= 0:
                 raise ValueError("maxtasksperchild must be a positive int or None")
+        if sys.platform == "win32" and processes > 61:
+            processes = 61
+            warnings.warn("Limiting number of processes in multiprocessing.Pool to 61, because of _winapi.WaitForMultipleObjects limit")

         if initializer is not None and not callable(initializer):
             raise TypeError('initializer must be a callable')
