diff --git a/Lib/sysconfig/__init__.py b/Lib/sysconfig/__init__.py
index 510c7b9..f368cf6 100644
--- a/Lib/sysconfig/__init__.py
+++ b/Lib/sysconfig/__init__.py
@@ -360,7 +360,12 @@ def _init_posix(vars):
     else:
         _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
     build_time_vars = _temp.build_time_vars
-    vars.update(build_time_vars)
+
+    prefix = build_time_vars.get('prefix', sys.prefix)
+    for k, v in build_time_vars.items():
+        if type(v) == type(''):
+            v = v.replace(prefix, sys.prefix)
+        vars[k] = v

 def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
