diff --git a/Tools/build/check_extension_modules.py b/Tools/build/check_extension_modules.py
index e0c7a92..bff7d8d 100644
--- a/Tools/build/check_extension_modules.py
+++ b/Tools/build/check_extension_modules.py
@@ -244,6 +244,7 @@ class ModuleChecker:
             )
             print_three_column(self.failed_on_import)
             print()
+            sys.exit(1)
 
         if any(
             modinfo.name == "_ssl" for modinfo in self.missing + self.failed_on_import
@@ -253,6 +254,7 @@ class ModuleChecker:
             if sysconfig.get_config_var("OPENSSL_LDFLAGS"):
                 print("Custom linker flags may require --with-openssl-rpath=auto")
             print()
+            sys.exit(1)
 
         disabled = len(self.disabled_configure) + len(self.disabled_setup)
         print(
