diff --git a/setup.py b/setup.py
index ad8fb81..28a7c6a 100644
--- a/setup.py
+++ b/setup.py
@@ -577,6 +577,7 @@ class PyBuildExt(build_ext):
             print("Failed to build these modules:")
             print_three_column(failed)
             print()
+            sys.exit(1)
 
         if self.failed_on_import:
             failed = self.failed_on_import[:]
@@ -585,6 +586,7 @@ class PyBuildExt(build_ext):
                   " but were removed because they could not be imported:")
             print_three_column(failed)
             print()
+            sys.exit(1)
 
         if any('_ssl' in l
                for l in (self.missing, self.failed, self.failed_on_import)):
@@ -594,6 +596,7 @@ class PyBuildExt(build_ext):
             if sysconfig.get_config_var("OPENSSL_LDFLAGS"):
                 print("Custom linker flags may require --with-openssl-rpath=auto")
             print()
+            sys.exit(1)
 
         if os.environ.get("PYTHONSTRICTEXTENSIONBUILD") and (
             self.failed or self.failed_on_import or self.missing
