diff --git a/Tools/build/check_extension_modules.py b/Tools/build/check_extension_modules.py
index 668db8d..38dc053 100644
--- a/Tools/build/check_extension_modules.py
+++ b/Tools/build/check_extension_modules.py
@@ -249,6 +249,7 @@ class ModuleChecker:
             )
             print_three_column(self.failed_on_import)
             print()
+            sys.exit(1)
 
         if any(
             modinfo.name == "_ssl" for modinfo in self.missing + self.failed_on_import
@@ -258,6 +259,7 @@ class ModuleChecker:
             if sysconfig.get_config_var("OPENSSL_LDFLAGS"):
                 print("Custom linker flags may require --with-openssl-rpath=auto")
             print()
+            sys.exit(1)
 
         disabled = len(self.disabled_configure) + len(self.disabled_setup)
         print(
