diff --git a/Python/pylifecycle.c b/Python/pylifecycle.c
index fbbc38b..99f175e 100644
--- a/Python/pylifecycle.c
+++ b/Python/pylifecycle.c
@@ -3097,6 +3097,10 @@ fatal_output_debug(const char *msg)
 
     OutputDebugStringW(L"Fatal Python error: ");
 
+    if (!msg) {
+        OutputDebugStringW(L"<message not set>\n");
+        return;
+    }
     msglen = strlen(msg);
     while (msglen) {
         size_t i;
