diff --git i/Lib/urllib/parse.py w/Lib/urllib/parse.py
index b7965fe..bd59852 100644
--- i/Lib/urllib/parse.py
+++ w/Lib/urllib/parse.py
@@ -470,7 +470,7 @@ def urlsplit(url, scheme='', allow_fragments=True):
         clear_cache()
     netloc = query = fragment = ''
     i = url.find(':')
-    if i > 0:
+    if i > 0 and url[0].isascii() and url[0].isalpha():
         for c in url[:i]:
             if c not in scheme_chars:
                 break
