diff --git a/cmake/pcl_find_boost.cmake b/cmake/pcl_find_boost.cmake
index f9e0b28..e609496 100644
--- a/cmake/pcl_find_boost.cmake
+++ b/cmake/pcl_find_boost.cmake
@@ -22,7 +22,7 @@ if(CMAKE_CXX_STANDARD MATCHES "14")
   set(BOOST_REQUIRED_MODULES filesystem iostreams system)
 else()
   # Optional boost modules
-  set(BOOST_OPTIONAL_MODULES filesystem serialization mpi)
+  set(BOOST_OPTIONAL_MODULES filesystem serialization)
   # Required boost modules
   set(BOOST_REQUIRED_MODULES iostreams system)
 endif()
