diff --git a/surface/include/pcl/surface/impl/concave_hull.hpp b/surface/include/pcl/surface/impl/concave_hull.hpp
index d769a65..b5d8410 100644
--- a/surface/include/pcl/surface/impl/concave_hull.hpp
+++ b/surface/include/pcl/surface/impl/concave_hull.hpp
@@ -133,7 +133,7 @@ pcl::ConcaveHull<PointInT>::performReconstruction (PointCloud &alpha_shape, std:
   EIGEN_ALIGN16 Eigen::Matrix3d eigen_vectors;
   pcl::eigen33 (covariance_matrix, eigen_vectors, eigen_values);
 
-  Eigen::Affine3d transform1;
+  EIGEN_ALIGN32 Eigen::Affine3d transform1;
   transform1.setIdentity ();
 
   // If no input dimension is specified, determine automatically
@@ -549,7 +549,7 @@ pcl::ConcaveHull<PointInT>::performReconstruction (PointCloud &alpha_shape, std:
   int curlong, totlong;
   qh_memfreeshort (qh, &curlong, &totlong);
 
-  Eigen::Affine3d transInverse = transform1.inverse ();
+  EIGEN_ALIGN32 Eigen::Affine3d transInverse = transform1.inverse ();
   pcl::transformPointCloud (alpha_shape, alpha_shape, transInverse);
   xyz_centroid[0] = - xyz_centroid[0];
   xyz_centroid[1] = - xyz_centroid[1];
