diff --git c/sphinx/locale/__init__.py i/sphinx/locale/__init__.py
index 4847c08..3764dc8 100644
--- c/sphinx/locale/__init__.py
+++ i/sphinx/locale/__init__.py
@@ -29,6 +29,10 @@ class _TranslationProxy(UserString, object):
     This also inherits from object to make the __new__ method work.
     """
     __slots__ = ('_func', '_args')
+    
+    # mevis: define __hash__ as None to prevent the following py3k warning:
+    # DeprecationWarning: Overriding __eq__ blocks inheritance of __hash__ in 3.x
+    __hash__ = None
 
     def __new__(cls, func, *args):
         if not args:
