diff --git i/zconf.h w/zconf.h
index bf977d3..75bf167 100644
--- i/zconf.h
+++ w/zconf.h
@@ -437,11 +437,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)
 #  define Z_HAVE_STDARG_H
 #endif
 
diff --git i/zconf.h.cmakein w/zconf.h.cmakein
index 247ba24..c1c7261 100644
--- i/zconf.h.cmakein
+++ w/zconf.h.cmakein
@@ -439,11 +439,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)
 #  define Z_HAVE_STDARG_H
 #endif
 
diff --git i/zconf.h.in w/zconf.h.in
index bf977d3..75bf167 100644
--- i/zconf.h.in
+++ w/zconf.h.in
@@ -437,11 +437,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)
 #  define Z_HAVE_STDARG_H
 #endif
 
