diff --git a/zconf.h b/zconf.h
index fb76ffe..f6f5e73 100644
--- i/zconf.h
+++ w/zconf.h
@@ -441,11 +441,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)
 #  define Z_HAVE_STDARG_H
 #endif
 
diff --git i/zconf.h.cmakein w/zconf.h.cmakein
index 310c439..82e38b2 100644
--- i/zconf.h.cmakein
+++ w/zconf.h.cmakein
@@ -443,11 +443,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)
 #  define Z_HAVE_STDARG_H
 #endif
 
diff --git i/zconf.h.in w/zconf.h.in
index fb76ffe..f6f5e73 100644
--- i/zconf.h.in
+++ w/zconf.h.in
@@ -441,11 +441,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_UNISTD_H) && (1-HAVE_UNISTD_H-1 != 0)
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if defined(HAVE_STDARG_H) && (1-HAVE_STDARG_H-1 != 0)
 #  define Z_HAVE_STDARG_H
 #endif

