diff --git c/CMakeLists.txt i/CMakeLists.txt
index dbb9bcf..8beb95a 100644
--- c/CMakeLists.txt
+++ i/CMakeLists.txt
@@ -438,7 +438,7 @@ endif()
 
 add_subdirectory(Eigen)
 
-add_subdirectory(doc EXCLUDE_FROM_ALL)
+#add_subdirectory(doc EXCLUDE_FROM_ALL)
 
 option(BUILD_TESTING "Enable creation of Eigen tests." ON)
 if(BUILD_TESTING)
@@ -447,7 +447,7 @@ if(BUILD_TESTING)
   if(EIGEN_LEAVE_TEST_IN_ALL_TARGET)
     add_subdirectory(test) # can't do EXCLUDE_FROM_ALL here, breaks CTest
   else()
-    add_subdirectory(test EXCLUDE_FROM_ALL)
+    #add_subdirectory(test EXCLUDE_FROM_ALL)
   endif()
 endif()
 
@@ -455,8 +455,8 @@ if(EIGEN_LEAVE_TEST_IN_ALL_TARGET)
   add_subdirectory(blas)
   add_subdirectory(lapack)
 else()
-  add_subdirectory(blas EXCLUDE_FROM_ALL)
-  add_subdirectory(lapack EXCLUDE_FROM_ALL)
+  #add_subdirectory(blas EXCLUDE_FROM_ALL)
+  #add_subdirectory(lapack EXCLUDE_FROM_ALL)
 endif()
 
 # add SYCL
@@ -468,18 +468,18 @@ endif()
 
 add_subdirectory(unsupported)
 
-add_subdirectory(demos EXCLUDE_FROM_ALL)
+#add_subdirectory(demos EXCLUDE_FROM_ALL)
 
 # must be after test and unsupported, for configuring buildtests.in
-add_subdirectory(scripts EXCLUDE_FROM_ALL)
+#add_subdirectory(scripts EXCLUDE_FROM_ALL)
 
 # TODO: consider also replacing EIGEN_BUILD_BTL by a custom target "make btl"?
 if(EIGEN_BUILD_BTL)
-  add_subdirectory(bench/btl EXCLUDE_FROM_ALL)
+  #add_subdirectory(bench/btl EXCLUDE_FROM_ALL)
 endif(EIGEN_BUILD_BTL)
 
 if(NOT WIN32)
-  add_subdirectory(bench/spbench EXCLUDE_FROM_ALL)
+  #add_subdirectory(bench/spbench EXCLUDE_FROM_ALL)
 endif(NOT WIN32)
 
 configure_file(scripts/cdashtesting.cmake.in cdashtesting.cmake @ONLY)
