diff --git c/qtbase/src/widgets/widgets/qtabwidget.cpp i/qtbase/src/widgets/widgets/qtabwidget.cpp
index 8a848554e..bd1326073 100644
--- c/qtbase/src/widgets/widgets/qtabwidget.cpp
+++ i/qtbase/src/widgets/widgets/qtabwidget.cpp
@@ -258,11 +258,9 @@ void QTabWidgetPrivate::init()
 
 bool QTabWidget::hasHeightForWidth() const
 {
-    Q_D(const QTabWidget);
-    bool has = d->size_policy.hasHeightForWidth();
-    if (!has && d->stack)
-        has = d->stack->hasHeightForWidth();
-    return has;
+    // Qt 4.8 added heightForWidth to QTabWidget, but this seems to be very buggy
+    // and caused some Tabs in MeVisLab to get invisible if they contained QLabels with heightForWidth.
+    return false;
 }
 
 /*!
