diff --git c/qtbase/src/gui/image/qicon.cpp i/qtbase/src/gui/image/qicon.cpp
index c362d0dc3..d4e01661a 100644
--- c/qtbase/src/gui/image/qicon.cpp
+++ i/qtbase/src/gui/image/qicon.cpp
@@ -900,7 +900,16 @@ QPixmap QIcon::pixmap(QWindow *window, const QSize &size, Mode mode, State state
     // Try get a pixmap that is big enough to be displayed at device pixel resolution.
     QIconEngine::ScaledPixmapArgument scalePixmapArg = { size * devicePixelRatio, mode, state, devicePixelRatio, QPixmap() };
     d->engine->virtual_hook(QIconEngine::ScaledPixmapHook, reinterpret_cast<void*>(&scalePixmapArg));
+#ifdef Q_OS_DARWIN
+    qreal sizeRatio = 0.5 * (qreal((size * devicePixelRatio).width()) / qreal(scalePixmapArg.pixmap.width()) +
+                             qreal((size * devicePixelRatio).height() / qreal(scalePixmapArg.pixmap.height())));
+    if (sizeRatio > 1.0 && sizeRatio < 2.0)
+        scalePixmapArg.pixmap.setDevicePixelRatio(devicePixelRatio);
+    else
+        scalePixmapArg.pixmap.setDevicePixelRatio(d->pixmapDevicePixelRatio(devicePixelRatio, size, scalePixmapArg.pixmap.size()));
+#else
     scalePixmapArg.pixmap.setDevicePixelRatio(d->pixmapDevicePixelRatio(devicePixelRatio, size, scalePixmapArg.pixmap.size()));
+#endif
     return scalePixmapArg.pixmap;
 }
 
