diff --git c/qtbase/src/opengl/qglshaderprogram.cpp i/qtbase/src/opengl/qglshaderprogram.cpp
index 545df8fa4..34f8cb2aa 100644
--- c/qtbase/src/opengl/qglshaderprogram.cpp
+++ i/qtbase/src/opengl/qglshaderprogram.cpp
@@ -952,22 +952,24 @@ bool QGLShaderProgram::link()
     value = 0;
     d->glfuncs->glGetProgramiv(program, GL_LINK_STATUS, &value);
     d->linked = (value != 0);
-    value = 0;
-    d->glfuncs->glGetProgramiv(program, GL_INFO_LOG_LENGTH, &value);
-    d->log = QString();
-    if (value > 1) {
-        char *logbuf = new char [value];
-        GLint len;
-        d->glfuncs->glGetProgramInfoLog(program, value, &len, logbuf);
-        d->log = QString::fromLatin1(logbuf);
-        QString name = objectName();
-        if (!d->linked) {
-            if (name.isEmpty())
-                qWarning() << "QGLShader::link:" << d->log;
-            else
-                qWarning() << "QGLShader::link[" << name << "]:" << d->log;
+    if (!d->linked) {
+        value = 0;
+        d->glfuncs->glGetProgramiv(program, GL_INFO_LOG_LENGTH, &value);
+        d->log = QString();
+        if (value > 1) {
+            char *logbuf = new char [value];
+            GLint len;
+            d->glfuncs->glGetProgramInfoLog(program, value, &len, logbuf);
+            d->log = QString::fromLatin1(logbuf);
+            QString name = objectName();
+            if (!d->linked) {
+                if (name.isEmpty())
+                    qWarning() << "QGLShader::link:" << d->log;
+                else
+                    qWarning() << "QGLShader::link[" << name << "]:" << d->log;
+            }
+            delete [] logbuf;
         }
-        delete [] logbuf;
     }
     return d->linked;
 }
