diff --git i/qtwebengine/src/core/config/common.pri w/qtwebengine/src/core/config/common.pri
index 42cf445dff..20d16c54fa 100644
--- i/qtwebengine/src/core/config/common.pri
+++ w/qtwebengine/src/core/config/common.pri
@@ -69,18 +69,13 @@ CONFIG(release, debug|release):!isDeveloperBuild() {
 
 CONFIG(release, debug|release) {
     gn_args += is_debug=false
-    force_debug_info {
-        # Level 1 is not enough to generate all Chromium debug symbols on Windows
-        msvc: gn_args += symbol_level=2
-        else: gn_args += symbol_level=1
-    } else {
-        gn_args += symbol_level=0
-    }
+    gn_args += symbol_level=0
 }
 
 CONFIG(debug, debug|release) {
     gn_args += is_debug=true
     gn_args += use_debug_fission=false
+    gn_args += symbol_level=0
 }
 
 !webcore_debug: gn_args += remove_webcore_debug_symbols=true
