diff --git c/qtbase/src/widgets/styles/qstyle.qrc i/qtbase/src/widgets/styles/qstyle.qrc
index d3511ee75..fa129a870 100644
--- c/qtbase/src/widgets/styles/qstyle.qrc
+++ i/qtbase/src/widgets/styles/qstyle.qrc
@@ -167,9 +167,13 @@
     </qresource>
     <qresource prefix="/qt-project.org/styles/macstyle">
         <file alias="images/closedock-16.png">images/closedock-macstyle-16.png</file>
+        <file alias="images/closedock-16@2x.png">images/closedock-macstyle-16@2x.png</file>
         <file alias="images/closedock-down-16.png">images/closedock-down-macstyle-16.png</file>
+        <file alias="images/closedock-down-16@2x.png">images/closedock-down-macstyle-16@2x.png</file>
         <file alias="images/dockdock-16.png">images/dockdock-macstyle-16.png</file>
+        <file alias="images/dockdock-16@2x.png">images/dockdock-macstyle-16@2x.png</file>
         <file alias="images/dockdock-down-16.png">images/dockdock-down-macstyle-16.png</file>
+        <file alias="images/dockdock-down-16@2x.png">images/dockdock-down-macstyle-16@2x.png</file>
         <file alias="images/toolbar-ext.png">images/toolbar-ext-macstyle.png</file>
         <file alias="images/toolbar-ext@2x.png">images/toolbar-ext-macstyle@2x.png</file>
     </qresource>
