diff --git i/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h w/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
index affd46750b..8f21b84a76 100644
--- i/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
+++ w/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
@@ -328,6 +328,11 @@ typedef NSString* VNImageOption NS_STRING_ENUM;
 @property(readonly, nonatomic, strong) VNFaceLandmarks2D* landmarks;
 @end
 
+#else
+
+// MeVis: Chromium seems to be missing the import for 10.13 and above
+#import <Vision/Vision.h>
+
 #endif  // MAC_OS_X_VERSION_10_13
 // ----------------------------------------------------------------------------
 // The symbol for kCWSSIDDidChangeNotification is available in the
