diff --git c/qtwebengine/src/3rdparty/chromium/qtwebengine/browser/BUILD.gn i/qtwebengine/src/3rdparty/chromium/qtwebengine/browser/BUILD.gn
index 302445d17b..e5ac8d0c3c 100644
--- c/qtwebengine/src/3rdparty/chromium/qtwebengine/browser/BUILD.gn
+++ i/qtwebengine/src/3rdparty/chromium/qtwebengine/browser/BUILD.gn
@@ -48,6 +48,7 @@ service_manifest("qtwebengine_content_browser_manifest_overlay") {
 
 group("qtwebengine_content_manifest_overlays") {
   deps = [
+    "//components/spellcheck:buildflags",
     ":qtwebengine_content_packaged_services_manifest_overlay",
     ":qtwebengine_content_renderer_manifest_overlay",
     ":qtwebengine_content_browser_manifest_overlay"
@@ -71,6 +72,7 @@ service_manifest("qtwebengine_content_browser_manifest") {
 
 group("service_manifests") {
   deps = [
+    "//components/spellcheck:buildflags",
     ":qtwebengine_content_renderer_manifest",
     ":qtwebengine_content_browser_manifest"
   ]
