Submodule qtwebengine contains modified content
diff --git i/qtwebengine/src/core/api/qwebenginemessagepumpscheduler.cpp w/qtwebengine/src/core/api/qwebenginemessagepumpscheduler.cpp
index 34cbc49..69f54c0 100644
--- i/qtwebengine/src/core/api/qwebenginemessagepumpscheduler.cpp
+++ w/qtwebengine/src/core/api/qwebenginemessagepumpscheduler.cpp
@@ -65,8 +65,10 @@ void QWebEngineMessagePumpScheduler::scheduleDelayedWork(int delay)
 
 void QWebEngineMessagePumpScheduler::timerEvent(QTimerEvent *ev)
 {
-    Q_ASSERT(!ev->timerId() || m_timerId == ev->timerId());
-    killTimer(m_timerId);
-    m_timerId = 0;
-    m_callback();
+    if (!ev->timerId() || m_timerId == ev->timerId())
+    {
+        killTimer(m_timerId);
+        m_timerId = 0;
+        m_callback();
+    }
 }
