diff --git a/qtbase/src/3rdparty/libpng/png.c b/qtbase/src/3rdparty/libpng/png.c
index 757c755f9..8982f3dba 100644
--- a/qtbase/src/3rdparty/libpng/png.c
+++ b/qtbase/src/3rdparty/libpng/png.c
@@ -2388,8 +2388,10 @@ png_compare_ICC_profile_with_sRGB(png_const_structrp png_ptr,
                       * discourage their use, skip the 'have_md5' warning below,
                       * which is made irrelevant by this error.
                       */
-                     png_chunk_report(png_ptr, "known incorrect sRGB profile",
-                         PNG_CHUNK_ERROR);
+                     // MeVis: disabled because many PNGs have these incorrect profiles
+                     // and Qt does not make use of them.
+                     //png_chunk_report(png_ptr, "known incorrect sRGB profile",
+                     //   PNG_CHUNK_ERROR);
                   }
 
                   /* Warn that this being done; this isn't even an error since
