diff --git a/qtbase/src/gui/opengl/qopenglvertexarrayobject.cpp b/qtbase/src/gui/opengl/qopenglvertexarrayobject.cpp
index 5a468bc6dd..f7085720d3 100644
--- a/qtbase/src/gui/opengl/qopenglvertexarrayobject.cpp
+++ b/qtbase/src/gui/opengl/qopenglvertexarrayobject.cpp
@@ -213,7 +213,7 @@ void QOpenGLVertexArrayObjectPrivate::destroy()
         // Before going through the effort of creating an offscreen surface
         // check that we are on the GUI thread because otherwise many platforms
         // will not able to create that offscreen surface.
-        if (QThread::currentThread() != guiThread) {
+        if (!qGuiApp || QThread::currentThread() != guiThread) {
             ctx = nullptr;
         } else {
             // Cannot just make the current surface current again with another context.
