diff --git a/qtbase/src/widgets/itemviews/qabstractitemview.cpp b/qtbase/src/widgets/itemviews/qabstractitemview.cpp
index 5e65c59796..03bfaaec76 100644
--- a/qtbase/src/widgets/itemviews/qabstractitemview.cpp
+++ b/qtbase/src/widgets/itemviews/qabstractitemview.cpp
@@ -2682,7 +2682,10 @@ bool QAbstractItemView::edit(const QModelIndex &index, EditTrigger trigger, QEve
     if (!d->shouldEdit(trigger, d->model->buddy(index)))
         return false;
 
-    if (d->delayedEditing.isActive())
+    if (trigger == AllEditTriggers) {
+        // editing is enforced now
+        d->delayedEditing.stop();
+    } else if (d->delayedEditing.isActive())
         return false;
 
     // we will receive a mouseButtonReleaseEvent after a
