diff --git i/qtsvg/src/svg/qsvgfont_p.h w/qtsvg/src/svg/qsvgfont_p.h
index fd0a3fab8..a29b8a9b2 100644
--- i/qtsvg/src/svg/qsvgfont_p.h
+++ w/qtsvg/src/svg/qsvgfont_p.h
@@ -74,7 +74,8 @@ public:
 class Q_SVG_PRIVATE_EXPORT QSvgFont : public QSvgRefCounted
 {
 public:
-    QSvgFont(qreal horizAdvX);
+    static constexpr qreal DEFAULT_UNITS_PER_EM = 1000;
+    QSvgFont(qreal horizAdvX);

     void setFamilyName(const QString &name);
     QString familyName() const;
@@ -86,9 +87,7 @@ public:
     void draw(QPainter *p, const QPointF &point, const QString &str, qreal pixelSize, Qt::Alignment alignment) const;
 public:
     QString m_familyName;
-    qreal m_unitsPerEm;
-    qreal m_ascent;
-    qreal m_descent;
+    qreal m_unitsPerEm = DEFAULT_UNITS_PER_EM;
     qreal m_horizAdvX;
     QHash<QChar, QSvgGlyph> m_glyphs;
 };
diff --git i/qtsvg/src/svg/qsvghandler.cpp w/qtsvg/src/svg/qsvghandler.cpp
index b2227b6a3..f4a00e32e 100644
--- i/qtsvg/src/svg/qsvghandler.cpp
+++ w/qtsvg/src/svg/qsvghandler.cpp
@@ -2666,7 +2666,7 @@ static bool parseFontFaceNode(QSvgStyleProperty *parent,

     qreal unitsPerEm = toDouble(unitsPerEmStr);
     if (!unitsPerEm)
-        unitsPerEm = 1000;
+        unitsPerEm = QSvgFont::DEFAULT_UNITS_PER_EM;

     if (!name.isEmpty())
         font->setFamilyName(name);
