diff --git i/qtbase/src/widgets/itemviews/qabstractitemview.cpp w/qtbase/src/widgets/itemviews/qabstractitemview.cpp
index b45a7bf704..eef31661e3 100644
--- i/qtbase/src/widgets/itemviews/qabstractitemview.cpp
+++ w/qtbase/src/widgets/itemviews/qabstractitemview.cpp
@@ -2691,7 +2691,10 @@ bool QAbstractItemView::edit(const QModelIndex &index, EditTrigger trigger, QEve
     if (!d->shouldEdit(trigger, d->model->buddy(index)))
         return false;
 
-    if (d->delayedEditing.isActive())
+    if (trigger == AllEditTriggers) {
+        // editing is enforced now
+        d->delayedEditing.stop();
+    } else if (d->delayedEditing.isActive())
         return false;
 
     // we will receive a mouseButtonReleaseEvent after a
