diff --git i/qtbase/mkspecs/features/qt_functions.prf w/qtbase/mkspecs/features/qt_functions.prf
index 7777e61..c6ef880 100644
--- i/qtbase/mkspecs/features/qt_functions.prf
+++ w/qtbase/mkspecs/features/qt_functions.prf
@@ -13,6 +13,12 @@ defineReplace(qtPlatformTargetSuffix) {
             }
         }
     }
+    linux {
+        CONFIG(debug, debug|release) {
+            !debug_and_release|build_pass: \
+                return($${suffix}d)
+        }
+    }
     darwin {
         CONFIG(debug, debug|release) {
             !debug_and_release|build_pass: \
diff --git i/qtbase/mkspecs/features/qt_helper_lib.prf w/qtbase/mkspecs/features/qt_helper_lib.prf
index 2d67a30..7bb8ade 100644
--- i/qtbase/mkspecs/features/qt_helper_lib.prf
+++ w/qtbase/mkspecs/features/qt_helper_lib.prf
@@ -61,7 +61,7 @@ win32|CONFIG(static, static|shared) {
     !single_android_abi:android {
         MODULE_PRI_CONT += "QMAKE_LIBS_$${ucmodule} ="
     } else: if(msvc|qtConfig(debug_and_release)): {
-        win32: \
+        win32|linux: \
             MODULE_DEBUG_LIBS = $$DESTDIR/$$prefix$${TARGET}d.$$suffix
         else: darwin: \
             MODULE_DEBUG_LIBS = $$DESTDIR/$$prefix$${TARGET}_debug.$$suffix
diff --git i/qtdeclarative/src/qml/qml/qqmlimport.cpp w/qtdeclarative/src/qml/qml/qqmlimport.cpp
index 10c6c41..0e0df27 100644
--- i/qtdeclarative/src/qml/qml/qqmlimport.cpp
+++ w/qtdeclarative/src/qml/qml/qqmlimport.cpp
@@ -2045,8 +2045,13 @@ QString QQmlImportDatabase::resolvePlugin(QQmlTypeLoader *typeLoader,
 # if defined(Q_OS_ANDROID)
         QStringLiteral(LIBS_SUFFIX),
 # endif
+# ifdef QT_DEBUG
+        QLatin1String("d.so"), // try a qmake-style debug build first
         QLatin1String(".so")
-
+# else
+        QLatin1String(".so"),
+        QLatin1String("d.so") // try a qmake-style debug build after
+# endif
     };
 #endif
 
