diff --git i/qtbase/src/widgets/kernel/qwidget.cpp w/qtbase/src/widgets/kernel/qwidget.cpp
index e94520021..a453f793d 100644
--- i/qtbase/src/widgets/kernel/qwidget.cpp
+++ w/qtbase/src/widgets/kernel/qwidget.cpp
@@ -7289,8 +7289,9 @@ bool QWidget::restoreGeometry(const QByteArray &geometry)
     // high DPI scaling or different levels of DPI awareness.
     if (restoredScreenWidth) {
         const qreal factor = qreal(restoredScreenWidth) / screenWidthF;
-        if (factor < 0.8 || factor > 1.25)
-            return false;
+        if (!maximized && !fullScreen)
+            if (factor < 0.8 || factor > 1.25)
+                return false;
     } else {
         // Saved by Qt 5.3 and earlier, try to prevent too large windows
         // unless the size will be adapted by maximized or fullscreen.
