diff --git i/src/PythonQtPythonInclude.h w/src/PythonQtPythonInclude.h
index c2b061f..2057c2f 100644
--- i/src/PythonQtPythonInclude.h
+++ w/src/PythonQtPythonInclude.h
@@ -33,19 +33,6 @@
 #ifndef __PythonQtPythonInclude_h
 #define __PythonQtPythonInclude_h
 
-// Undefine macros that Python.h defines to avoid redefinition warning.
-#ifdef _POSIX_C_SOURCE
-#  undef _POSIX_C_SOURCE
-#endif
-
-#ifdef _POSIX_THREADS
-#  undef _POSIX_THREADS
-#endif
-
-#ifdef _XOPEN_SOURCE
-#  undef _XOPEN_SOURCE
-#endif
-
 // Undefine Qt keywords that conflict with Python headers
 #ifdef slots
 #undef slots
