diff --git i/meson.build w/meson.build
index 1550bb2..c6f4454 100644
--- i/meson.build
+++ w/meson.build
@@ -1561,7 +1561,6 @@ else
 endif
 
 dep_zlib = dependency('zlib', version : '>= 1.2.3',
-                      fallback : ['zlib', 'zlib_dep'],
                       required : get_option('zlib'))
 if dep_zlib.found()
   pre_args += '-DHAVE_ZLIB'
@@ -1605,7 +1604,7 @@ endif
 if host_machine.system() == 'darwin'
   dep_expat = meson.get_compiler('c').find_library('expat')
 elif host_machine.system() != 'windows'
-  dep_expat = dependency('expat', fallback : ['expat', 'expat_dep'],
+  dep_expat = dependency('expat',
                          required: not with_platform_android)
 else
   dep_expat = null_dep
@@ -1790,7 +1789,6 @@ if _llvm != 'disabled'
     ),
     static : not _shared_llvm,
     method : _llvm_method,
-    fallback : ['llvm', 'dep_llvm'],
     include_type : 'system',
   )
   with_llvm = dep_llvm.found()
