diff --git a/src/core/api/qwebenginemessagepumpscheduler.cpp b/src/core/api/qwebenginemessagepumpscheduler.cpp
index 860696c374..8ec05932d0 100644
--- a/src/core/api/qwebenginemessagepumpscheduler.cpp
+++ b/src/core/api/qwebenginemessagepumpscheduler.cpp
@@ -29,10 +29,12 @@ void QWebEngineMessagePumpScheduler::scheduleDelayedWork(int delay)
 
 void QWebEngineMessagePumpScheduler::timerEvent(QTimerEvent *ev)
 {
-    Q_ASSERT(!ev->timerId() || m_timerId == ev->timerId());
-    killTimer(m_timerId);
-    m_timerId = 0;
-    m_callback();
+   if (!ev->timerId() || m_timerId == ev->timerId())
+   {
+       killTimer(m_timerId);
+       m_timerId = 0;
+       m_callback();
+   }
 }
 
 #include "moc_qwebenginemessagepumpscheduler_p.cpp"
