diff --git a/src/core/printing/printer_worker.cpp b/src/core/printing/printer_worker.cpp
index 3d1c1f4ec8..f717f84da1 100644
--- a/src/core/printing/printer_worker.cpp
+++ b/src/core/printing/printer_worker.cpp
@@ -20,7 +20,7 @@ PrinterWorker::~PrinterWorker() { }
 
 void PrinterWorker::print()
 {
-    if (!m_data->size()) {
+    if (m_data.isNull() || !m_data->size()) {
         qWarning("Failed to print: Print result data is empty.");
         Q_EMIT resultReady(false);
         return;
