diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 2b71f80..692a55d 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -639,6 +639,10 @@ if(COIN_BUG_699)
    set_property(TARGET WebEngineCore PROPERTY CXX_LINKER_LAUNCHER ${PROJECT_BINARY_DIR}/linker_ulimit.sh)
 endif()
 
+if (UNIX AND NOT APPLE)
+    target_link_options(WebEngineCore PRIVATE -Wl,-z,noexecstack)
+endif()
+
 ##
 #  SANDBOX
 ##
