Avoid crash if anything goes wrong while printing
---
diff --git a/src/core/printing/printer_worker.cpp b/src/core/printing/printer_worker.cpp
index da71656..40a7e5a 100644
--- a/src/core/printing/printer_worker.cpp
+++ b/src/core/printing/printer_worker.cpp
@@ -19,7 +19,7 @@ PrinterWorker::~PrinterWorker() { }
 
 void PrinterWorker::print()
 {
-    if (!m_data->size()) {
+    if (m_data.isNull() || !m_data->size()) {
         qWarning("Failed to print: Print result data is empty.");
         Q_EMIT resultReady(false);
         return;
