diff --git i/include/llvm/Support/type_traits.h w/include/llvm/Support/type_traits.h
index 7b7d5d9..09f4ffd 100644
--- i/include/llvm/Support/type_traits.h
+++ w/include/llvm/Support/type_traits.h
@@ -176,7 +176,8 @@ class is_trivially_copyable {
       (has_deleted_copy_assign || has_trivial_copy_assign) &&
       (has_deleted_copy_constructor || has_trivial_copy_constructor);
 
-#ifdef HAVE_STD_IS_TRIVIALLY_COPYABLE
+// see https://developercommunity.visualstudio.com/content/problem/1144107/vs-167-breaks-is-trivially-copyable-for-stdpair.html
+#if defined(HAVE_STD_IS_TRIVIALLY_COPYABLE) && !defined(_MSC_VER)
   static_assert(value == std::is_trivially_copyable<T>::value,
                 "inconsistent behavior between llvm:: and std:: implementation of is_trivially_copyable");
 #endif
