diff --git i/include/llvm/Support/FormattedStream.h w/include/llvm/Support/FormattedStream.h
index 5f937cfa..4a253eca 100644
--- i/include/llvm/Support/FormattedStream.h
+++ w/include/llvm/Support/FormattedStream.h
@@ -145,7 +145,7 @@ public:
     return *this;
   }
 
-  raw_ostream &changeColor(enum Colors Color, bool Bold, bool BG) override {
+  raw_ostream &changeColor(Colors Color, bool Bold, bool BG) override {
     TheStream->changeColor(Color, Bold, BG);
     return *this;
   }
diff --git i/include/llvm/Support/raw_ostream.h w/include/llvm/Support/raw_ostream.h
index 8d289f7c..e3e6b2be 100644
--- i/include/llvm/Support/raw_ostream.h
+++ w/include/llvm/Support/raw_ostream.h
@@ -275,7 +275,7 @@ public:
   /// @param Bold bold/brighter text, default false
   /// @param BG if true change the background, default: change foreground
   /// @returns itself so it can be used within << invocations
-  virtual raw_ostream &changeColor(enum Colors Color, bool Bold = false,
+  virtual raw_ostream &changeColor(Colors Color, bool Bold = false,
                                    bool BG = false);
 
   /// Resets the colors to terminal defaults. Call this when you are done
