diff --git a/CMakeLists.txt b/CMakeLists.txt
index 00d01f4d..d0065f34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,29 @@ project(LLVM
 # Must go after project(..)
 include(GNUInstallDirs)
 
+set(_unused
+    ${CMAKE_EXPORT_NO_PACKAGE_REGISTRY}
+    ${CMAKE_INSTALL_BINDIR}
+    ${CMAKE_INSTALL_DATAROOTDIR}
+    ${CMAKE_INSTALL_INCLUDEDIR}
+    ${CMAKE_INSTALL_LIBDIR}
+    ${CMAKE_INSTALL_LIBEXECDIR}
+    ${CMAKE_INSTALL_OLDINCLUDEDIR}
+    ${CMAKE_INSTALL_SBINDIR}
+    ${CMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE}
+    ${CONAN_COMPILER}
+    ${CONAN_COMPILER_VERSION}
+    ${CONAN_CXX_FLAGS}
+    ${CONAN_C_FLAGS}
+    ${CONAN_EXPORTED}
+    ${CONAN_IN_LOCAL_CACHE}
+    ${CONAN_LIBCXX}
+    ${CONAN_SHARED_LINKER_FLAGS}
+)
+
+include(${CMAKE_BINARY_DIR}/../conanbuildinfo.cmake)
+conan_basic_setup()
+
 set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard to conform to")
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 if (CYGWIN)
