diff --git c/CDT/include/CDT.h i/CDT/include/CDT.h
index 475706c..ead110a 100644
--- c/CDT/include/CDT.h
+++ i/CDT/include/CDT.h
@@ -512,8 +512,7 @@ namespace CDT
 #ifdef CDT_CXX11_IS_SUPPORTED
 inline void shuffle_indices(std::vector<VertInd>& indices)
 {
-    std::random_device rd;
-    std::mt19937 g(rd());
+    std::mt19937 g(9001);
     std::shuffle(indices.begin(), indices.end(), g);
 }
 #else
-- 
