diff --git i/CDT/include/KDTree.h w/CDT/include/KDTree.h
index d484e3c..fbcf7e7 100644
--- i/CDT/include/KDTree.h
+++ w/CDT/include/KDTree.h
@@ -194,9 +194,9 @@ public:
             }
             else
             {
-                coord_type mid;
-                NodeSplitDirection::Enum newDir;
-                point_type newMin, newMax;
+                coord_type mid {};
+                NodeSplitDirection::Enum newDir {};
+                point_type newMin {}, newMax {};
                 calcSplitInfo(t.min, t.max, t.dir, mid, newDir, newMin, newMax);
 
                 const coord_type distToMid = t.dir == NodeSplitDirection::X
@@ -371,4 +371,4 @@ private:
 
 } // namespace KDTree
 
-#endif // header guard
\ No newline at end of file
+#endif // header guard
