diff --git c/coverage/html.py i/coverage/html.py
index b0c6164..dc18d64 100644
--- c/coverage/html.py
+++ i/coverage/html.py
@@ -173,7 +173,9 @@ class HtmlReporter(Reporter):
     def html_file(self, fr, analysis):
         """Generate an HTML file for one source file."""
         rootname = flat_rootname(fr.relative_filename())
-        html_filename = rootname + ".html"
+        filenamehasher = Hasher()
+        filenamehasher.update(rootname)
+        html_filename = filenamehasher.hexdigest() + ".html"
         html_path = os.path.join(self.directory, html_filename)
 
         # Get the numbers for this file.
