diff --git c/Source/CMakeLists.txt i/Source/CMakeLists.txt
index 9633765..f21c60e 100644
--- c/Source/CMakeLists.txt
+++ i/Source/CMakeLists.txt
@@ -72,7 +72,7 @@ SET(OpenIGTLink_SOURCES
   )

 SET(OpenIGTLink_INCLUDE_FILES)
-IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
+#IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
   LIST(APPEND OpenIGTLink_INCLUDE_FILES
     igtlutil/igtl_header.h
     igtlutil/igtl_image.h
@@ -117,7 +117,7 @@ IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
     igtlWindows.h
     igtlCommon.h
   )
-ENDIF()
+#ENDIF()

 # Add support for OpenIGTLink version 2
 IF (${OpenIGTLink_PROTOCOL_VERSION} GREATER "1" )
@@ -150,7 +150,7 @@ IF (${OpenIGTLink_PROTOCOL_VERSION} GREATER "1" )
     igtlBindMessage.cxx
     igtlNDArrayMessage.cxx
     )
-  IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
+  #IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
     LIST(APPEND OpenIGTLink_INCLUDE_FILES
       igtlutil/igtl_colortable.h
       igtlutil/igtl_imgmeta.h
@@ -180,7 +180,7 @@ IF (${OpenIGTLink_PROTOCOL_VERSION} GREATER "1" )
       igtlBindMessage.h
       igtlNDArrayMessage.h
       )
-  ENDIF()
+  #ENDIF()
 ENDIF()

 # Add support for OpenIGTLink version 3
@@ -191,14 +191,14 @@ IF( ${OpenIGTLink_PROTOCOL_VERSION} GREATER "2" )
     igtlutil/igtl_command.c
     igtlutil/igtl_query.c
     )
-  IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
+  #IF( MSVC OR ${CMAKE_GENERATOR} MATCHES "Xcode" )
     LIST(APPEND OpenIGTLink_INCLUDE_FILES
       igtlCommandMessage.h
       igtlQueryMessage.h
       igtlutil/igtl_command.h
       igtlutil/igtl_query.h
       )
-  ENDIF()
+  #ENDIF()
 ENDIF()

 ADD_LIBRARY(OpenIGTLink ${OpenIGTLink_SOURCES} ${OpenIGTLink_INCLUDE_FILES})
