diff --git a/src/google/protobuf/message_lite.h b/src/google/protobuf/message_lite.h
index 25264ef..bf5d571 100644
--- a/src/google/protobuf/message_lite.h
+++ b/src/google/protobuf/message_lite.h
@@ -292,11 +292,19 @@ struct MessageTraitsImpl {
 template <typename T>
 using MessageTraits = decltype(MessageTraitsImpl::value<T>);
 
+#ifdef __CUDACC__
+struct EnumTraitsImpl {
+  template <typename T>
+  static std::enable_if_t<sizeof(T) != 0> value;
+};
+#else
 struct EnumTraitsImpl {
   struct Undefined;
   template <typename T>
   static Undefined value;
 };
+#endif
+
 template <typename T>
 using EnumTraits = decltype(EnumTraitsImpl::value<T>);
 
