diff --git c/BLAS/SRC/xerbla.c i/BLAS/SRC/xerbla.c
index 2d7baf5..1088d0f 100644
--- c/BLAS/SRC/xerbla.c
+++ i/BLAS/SRC/xerbla.c
@@ -12,6 +12,7 @@
 
 #include "f2c.h"
 #include "blaswrap.h"
+#include <stdio.h>
 
 /* Table of constant values */
 
diff --git c/SRC/xerbla.c i/SRC/xerbla.c
index fb4a9d8..568552d 100644
--- c/SRC/xerbla.c
+++ i/SRC/xerbla.c
@@ -12,7 +12,7 @@
 
 #include "f2c.h"
 #include "blaswrap.h"
-#include "stdio.h"
+#include <stdio.h>
 
 /* Table of constant values */
 
