diff --git c/F2CLIBS/libf2c/f2c.h i/F2CLIBS/libf2c/f2c.h
index b94ee7c..b4af07e 100644
--- c/F2CLIBS/libf2c/f2c.h
+++ i/F2CLIBS/libf2c/f2c.h
@@ -154,6 +154,7 @@ struct Namelist {
 	};
 typedef struct Namelist Namelist;
 
+#ifdef BUILD_CLAPACK
 #define abs(x) ((x) >= 0 ? (x) : -(x))
 #define dabs(x) (doublereal)abs(x)
 #define min(a,b) ((a) <= (b) ? (a) : (b))
@@ -163,6 +164,7 @@ typedef struct Namelist Namelist;
 #define bit_test(a,b)	((a) >> (b) & 1)
 #define bit_clear(a,b)	((a) & ~((uinteger)1 << (b)))
 #define bit_set(a,b)	((a) |  ((uinteger)1 << (b)))
+#endif
 
 /* procedure parameter types for -A and -C++ */
 
diff --git c/INCLUDE/f2c.h i/INCLUDE/f2c.h
index b94ee7c..b4af07e 100644
--- c/INCLUDE/f2c.h
+++ i/INCLUDE/f2c.h
@@ -154,6 +154,7 @@ struct Namelist {
 	};
 typedef struct Namelist Namelist;
 
+#ifdef BUILD_CLAPACK
 #define abs(x) ((x) >= 0 ? (x) : -(x))
 #define dabs(x) (doublereal)abs(x)
 #define min(a,b) ((a) <= (b) ? (a) : (b))
@@ -163,6 +164,7 @@ typedef struct Namelist Namelist;
 #define bit_test(a,b)	((a) >> (b) & 1)
 #define bit_clear(a,b)	((a) & ~((uinteger)1 << (b)))
 #define bit_set(a,b)	((a) |  ((uinteger)1 << (b)))
+#endif
 
 /* procedure parameter types for -A and -C++ */
 
