diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93d61ec..521713d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -247,24 +247,24 @@ set_target_properties(
 	SOVERSION ${LIBXSLT_MAJOR_VERSION}
 )
 
-if(MSVC)
-	if(BUILD_SHARED_LIBS)
-		set_target_properties(
-			LibXslt
-			PROPERTIES
-			DEBUG_POSTFIX d
-		)
-	else()
-		set_target_properties(
-			LibXslt
-			PROPERTIES
-			DEBUG_POSTFIX sd
-			MINSIZEREL_POSTFIX s
-			RELEASE_POSTFIX s
-			RELWITHDEBINFO_POSTFIX s
-		)
-	endif()
-endif()
+# if(MSVC)
+# 	if(BUILD_SHARED_LIBS)
+# 		set_target_properties(
+# 			LibXslt
+# 			PROPERTIES
+# 			DEBUG_POSTFIX d
+# 		)
+# 	else()
+# 		set_target_properties(
+# 			LibXslt
+# 			PROPERTIES
+# 			DEBUG_POSTFIX sd
+# 			MINSIZEREL_POSTFIX s
+# 			RELEASE_POSTFIX s
+# 			RELWITHDEBINFO_POSTFIX s
+# 		)
+# 	endif()
+# endif()
 
 install(FILES ${LIBXSLT_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libxslt COMPONENT development)
 
@@ -340,24 +340,24 @@ set_target_properties(
 	SOVERSION ${LIBEXSLT_MAJOR_VERSION}
 )
 
-if(MSVC)
-	if(BUILD_SHARED_LIBS)
-		set_target_properties(
-			LibExslt
-			PROPERTIES
-			DEBUG_POSTFIX d
-		)
-	else()
-		set_target_properties(
-			LibExslt
-			PROPERTIES
-			DEBUG_POSTFIX sd
-			MINSIZEREL_POSTFIX s
-			RELEASE_POSTFIX s
-			RELWITHDEBINFO_POSTFIX s
-		)
-	endif()
-endif()
+# if(MSVC)
+# 	if(BUILD_SHARED_LIBS)
+# 		set_target_properties(
+# 			LibExslt
+# 			PROPERTIES
+# 			DEBUG_POSTFIX d
+# 		)
+# 	else()
+# 		set_target_properties(
+# 			LibExslt
+# 			PROPERTIES
+# 			DEBUG_POSTFIX sd
+# 			MINSIZEREL_POSTFIX s
+# 			RELEASE_POSTFIX s
+# 			RELWITHDEBINFO_POSTFIX s
+# 		)
+# 	endif()
+# endif()
 
 install(FILES ${LIBEXSLT_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libexslt COMPONENT development)
 
