diff --git a/CMakeLists.txt b/CMakeLists.txt
index b952d7b..c938701 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -425,24 +425,24 @@ set_target_properties(
     MACHO_CURRENT_VERSION "${LIBXML_MACHO_COMPAT}.${LIBXML_MICRO_VERSION}"
 )
 
-if(MSVC)
-    if(BUILD_SHARED_LIBS)
-        set_target_properties(
-            LibXml2
-            PROPERTIES
-            DEBUG_POSTFIX d
-        )
-    else()
-        set_target_properties(
-            LibXml2
-            PROPERTIES
-            DEBUG_POSTFIX sd
-            MINSIZEREL_POSTFIX s
-            RELEASE_POSTFIX s
-            RELWITHDEBINFO_POSTFIX s
-        )
-    endif()
-endif()
+# if(MSVC)
+#     if(BUILD_SHARED_LIBS)
+#         set_target_properties(
+#             LibXml2
+#             PROPERTIES
+#             DEBUG_POSTFIX d
+#         )
+#     else()
+#         set_target_properties(
+#             LibXml2
+#             PROPERTIES
+#             DEBUG_POSTFIX sd
+#             MINSIZEREL_POSTFIX s
+#             RELEASE_POSTFIX s
+#             RELWITHDEBINFO_POSTFIX s
+#         )
+#     endif()
+# endif()
 
 set(XML_SYSCONFDIR "${CMAKE_INSTALL_FULL_SYSCONFDIR}")
 
