diff --git c/CMakeLists.txt i/CMakeLists.txt
index dd3d8eb..a47fc32 100644
--- c/CMakeLists.txt
+++ i/CMakeLists.txt
@@ -824,7 +824,7 @@ endif()
 # -----------------------------------------------------------------------------
 # force off-tree build
 
-if(${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_BINARY_DIR})
+if(0)
   message(
     FATAL_ERROR
       "CMake generation is not allowed within the source directory! \
