diff --git i/cryptopp-cmake/CMakeLists.txt w/cryptopp-cmake/CMakeLists.txt
index b0ee7a2..d0ba2fd 100644
--- i/cryptopp-cmake/CMakeLists.txt
+++ w/cryptopp-cmake/CMakeLists.txt
@@ -192,7 +192,7 @@ include(cmake/FasterBuild.cmake)
 # If CMake is invoked with an explicit option (cryptopp_SOURCE_DIR), setting
 # the location for user-provided sources of crypto++, or if the automatic
 # download fails, we use that option value to find the sources.
-find_package(Git)
+# find_package(Git)   leads to crash because it assumed the temp conan directory is git controlled
 
 if(NOT CRYPTOPP_SOURCES)
   include(GetCryptoppSources)
