diff --git a/lib/matplotlib/backends/backend_qt.py b/lib/matplotlib/backends/backend_qt.py
index 475acbf..35bb245 100644
--- a/lib/matplotlib/backends/backend_qt.py
+++ b/lib/matplotlib/backends/backend_qt.py
@@ -652,6 +652,9 @@ class FigureManagerQT(FigureManagerBase):
 
     @classmethod
     def start_main_loop(cls):
+        if QtCore.QThread.currentThread().loopLevel():
+            # there is already an event loop running
+            return
         qapp = QtWidgets.QApplication.instance()
         if qapp:
             with _allow_interrupt_qt(qapp):
