diff --git a/src/ft2font.cpp b/src/ft2font.cpp
index 3fc9e8b..b58530f 100644
--- a/src/ft2font.cpp
+++ b/src/ft2font.cpp
@@ -227,7 +227,7 @@ int FT2Font::get_path_count()
         v_control = v_start;
 
         point = outline.points + first;
-        tags = outline.tags + first;
+        tags = reinterpret_cast<char*>(outline.tags + first);
         tag = FT_CURVE_TAG(tags[0]);
 
         // A contour cannot start with a cubic control point!
@@ -343,7 +343,7 @@ void FT2Font::get_path(double *outpoints, unsigned char *outcodes)
         v_control = v_start;
 
         point = outline.points + first;
-        tags = outline.tags + first;
+        tags = reinterpret_cast<char*>(outline.tags + first);
         tag = FT_CURVE_TAG(tags[0]);
 
         double x, y;
