diff --git i/src/_path.cpp w/src/_path.cpp
index f9c7771..ad49ca9 100644
--- i/src/_path.cpp
+++ w/src/_path.cpp
@@ -4,6 +4,7 @@
 #include "agg_py_transforms.h"
 #include "path_converters.h"
 
+#include <algorithm>
 #include <limits>
 #include <math.h>
 
diff --git i/src/ft2font.cpp w/src/ft2font.cpp
index c0756f2..62d257d 100644
--- i/src/ft2font.cpp
+++ w/src/ft2font.cpp
@@ -2,6 +2,7 @@
 
 #include "ft2font.h"
 #include "mplutils.h"
+#include <algorithm>
 #include <sstream>
 
 #include "file_compat.h"
