diff --git i/lib/matplotlib/backends/backend_qt4agg.py w/lib/matplotlib/backends/backend_qt4agg.py
index a597dc2..09deed0 100644
--- i/lib/matplotlib/backends/backend_qt4agg.py
+++ w/lib/matplotlib/backends/backend_qt4agg.py
@@ -8,7 +8,6 @@ import six
 
 import os  # not used
 import sys
-import ctypes
 import warnings
 
 import matplotlib
@@ -31,10 +30,6 @@ from matplotlib.cbook import mplDeprecation
 
 DEBUG = False
 
-_decref = ctypes.pythonapi.Py_DecRef
-_decref.argtypes = [ctypes.py_object]
-_decref.restype = None
-
 
 def new_figure_manager(num, *args, **kwargs):
     """
diff --git i/lib/matplotlib/backends/backend_qt5agg.py w/lib/matplotlib/backends/backend_qt5agg.py
index 9b21636..3dee24b 100644
--- i/lib/matplotlib/backends/backend_qt5agg.py
+++ w/lib/matplotlib/backends/backend_qt5agg.py
@@ -8,7 +8,6 @@ import six
 
 import os  # not used
 import sys
-import ctypes
 import warnings
 
 import matplotlib
@@ -32,10 +31,6 @@ from matplotlib.cbook import mplDeprecation
 
 DEBUG = False
 
-_decref = ctypes.pythonapi.Py_DecRef
-_decref.argtypes = [ctypes.py_object]
-_decref.restype = None
-
 
 def new_figure_manager(num, *args, **kwargs):
     """
@@ -92,8 +87,6 @@ class FigureCanvasQTAggBase(object):
             else:
                 stringBuffer = self.renderer._renderer.tostring_argb()
 
-            refcnt = sys.getrefcount(stringBuffer)
-
             # convert the Agg rendered image -> qImage
             qImage = QtGui.QImage(stringBuffer, self.renderer.width,
                                   self.renderer.height,
@@ -113,13 +106,7 @@ class FigureCanvasQTAggBase(object):
                 p.drawRect(x, y, w, h)
             p.end()
 
-            # This works around a bug in PySide 1.1.2 on Python 3.x,
-            # where the reference count of stringBuffer is incremented
-            # but never decremented by QImage.
-            # TODO: revert PR #1323 once the issue is fixed in PySide.
             del qImage
-            if refcnt != sys.getrefcount(stringBuffer):
-                _decref(stringBuffer)
         else:
             bbox = self.blitbox
             l, b, r, t = bbox.extents
