diff --git i/lib/matplotlib/__init__.py w/lib/matplotlib/__init__.py
index c172709..f04511d 100644
--- i/lib/matplotlib/__init__.py
+++ w/lib/matplotlib/__init__.py
@@ -750,42 +750,8 @@ def matplotlib_fname():
     - Lastly, it looks in `$MATPLOTLIBDATA/matplotlibrc` for a
       system-defined copy.
     """
-    if six.PY2:
-        cwd = os.getcwdu()
-    else:
-        cwd = os.getcwd()
-    fname = os.path.join(cwd, 'matplotlibrc')
-    if os.path.exists(fname):
-        return fname
-
-    if 'MATPLOTLIBRC' in os.environ:
-        path =  os.environ['MATPLOTLIBRC']
-        if os.path.exists(path):
-            fname = os.path.join(path, 'matplotlibrc')
-            if os.path.exists(fname):
-                return fname
-
-    configdir = _get_configdir()
-    if configdir is not None:
-        fname = os.path.join(configdir, 'matplotlibrc')
-        if os.path.exists(fname):
-            home = get_home()
-            if (sys.platform.startswith('linux') and
-                home is not None and
-                os.path.exists(os.path.join(
-                    home, '.matplotlib', 'matplotlibrc'))):
-                warnings.warn(
-                    "Found matplotlib configuration in ~/.matplotlib/. "
-                    "To conform with the XDG base directory standard, "
-                    "this configuration location has been deprecated "
-                    "on Linux, and the new location is now %s/matplotlib/. "
-                    "Please move your configuration there to ensure that "
-                    "matplotlib will continue to find it in the future." %
-                    _get_xdg_config_dir())
-                return os.path.join(
-                    home, '.matplotlib', 'matplotlibrc')
-            return fname
-
+    
+    # MeVisLab shortcut, we only support the built-in rc file
     path = get_data_path()  # guaranteed to exist or raise
     fname = os.path.join(path, 'matplotlibrc')
     if not os.path.exists(fname):
diff --git i/lib/matplotlib/mpl-data/matplotlibrc w/lib/matplotlib/mpl-data/matplotlibrc
index c72d6d7..0d1e2ca 100644
--- i/lib/matplotlib/mpl-data/matplotlibrc
+++ w/lib/matplotlib/mpl-data/matplotlibrc
@@ -40,7 +40,7 @@ backend      : qt4agg
 # If you are using the Qt4Agg backend, you can choose here
 # to use the PyQt4 bindings or the newer PySide bindings to
 # the underlying Qt4 toolkit.
-#backend.qt4 : PyQt4        # PyQt4 | PySide
+backend.qt4 : PythonQt
 
 # Note that this can be overridden by the environment variable
 # QT_API used by Enthought Tool Suite (ETS); valid values are
diff --git i/lib/matplotlib/rcsetup.py w/lib/matplotlib/rcsetup.py
index 65410fe..c30aefa 100644
--- i/lib/matplotlib/rcsetup.py
+++ w/lib/matplotlib/rcsetup.py
@@ -161,7 +161,7 @@ def validate_backend(s):
         return _validate_standard_backends(s)
 
 
-validate_qt4 = ValidateInStrings('backend.qt4', ['PyQt4', 'PySide', 'PyQt4v2'])
+validate_qt4 = ValidateInStrings('backend.qt4', ['PyQt4', 'PySide', 'PyQt4v2', 'PythonQt'])
 validate_qt5 = ValidateInStrings('backend.qt5', ['PyQt5'])
 
 
diff --git i/matplotlibrc.template w/matplotlibrc.template
index f742fe3..4a0d061 100644
--- i/matplotlibrc.template
+++ w/matplotlibrc.template
@@ -40,7 +40,7 @@ backend      : %(backend)s
 # If you are using the Qt4Agg backend, you can choose here
 # to use the PyQt4 bindings or the newer PySide bindings to
 # the underlying Qt4 toolkit.
-#backend.qt4 : PyQt4        # PyQt4 | PySide
+backend.qt4 : PythonQt        # PyQt4 | PySide
 
 # Note that this can be overridden by the environment variable
 # QT_API used by Enthought Tool Suite (ETS); valid values are
