From ee0e80e4b21fc3e5fb3308fdc9c838015fe89e30 Mon Sep 17 00:00:00 2001
From: Jan Plath <jan.plath@mevis.de>
Date: Wed, 17 Sep 2025 13:02:08 +0200
Subject: [PATCH 2/2] version3

---
 Rendering/OpenGL2/vtkGenericOpenGLRenderWindow.cxx | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Rendering/OpenGL2/vtkGenericOpenGLRenderWindow.cxx b/Rendering/OpenGL2/vtkGenericOpenGLRenderWindow.cxx
index 084f01c9..f552ad83 100644
--- a/Rendering/OpenGL2/vtkGenericOpenGLRenderWindow.cxx
+++ b/Rendering/OpenGL2/vtkGenericOpenGLRenderWindow.cxx
@@ -177,6 +177,7 @@ void vtkGenericOpenGLRenderWindow::SetIsCurrent(bool newValue)
 
 void vtkGenericOpenGLRenderWindow::OpenGLInit()
 {
+  this->OpenGLInitContext();
   auto state = this->GetState();
   state->Reset();
   state->Push();
@@ -188,7 +189,6 @@ void vtkGenericOpenGLRenderWindow::Render()
 {
   if (this->ReadyForRendering)
   {
-    vtkOpenGLCheckErrorMacro("error before running VTK rendering code");
     if (!this->Initialized)
     {
       this->OpenGLInit();
@@ -200,6 +200,7 @@ void vtkGenericOpenGLRenderWindow::Render()
     }
     else
     {
+      vtkOpenGLCheckErrorMacro("error before running VTK rendering code");
       // Query current GL state and store them
       auto state = this->GetState();
       state->Reset();
@@ -209,8 +210,8 @@ void vtkGenericOpenGLRenderWindow::Render()
 
       // Restore state to previous known value
       state->Pop();
+      vtkOpenGLCheckErrorMacro("error after running VTK rendering code");
     }
-    vtkOpenGLCheckErrorMacro("error after running VTK rendering code");
   }
 }
 
-- 
2.51.0.windows.1

