diff --git a/Rendering/OpenGL2/vtkOpenGLState.cxx b/Rendering/OpenGL2/vtkOpenGLState.cxx
index 5d2a85e4..dfd51461 100644
--- a/Rendering/OpenGL2/vtkOpenGLState.cxx
+++ b/Rendering/OpenGL2/vtkOpenGLState.cxx
@@ -1746,6 +1746,7 @@ void vtkOpenGLState::vtkglBlitFramebuffer(int srcX0, int srcY0, int srcX1, int s
   this->vtkglDisable(GL_SCISSOR_TEST);
 
   ::glBlitFramebuffer(srcX0, srcY0, srcX1, srcY1, dstX0, dstY0, dstX1, dstY1, mask, filter);
+  vtkCheckOpenGLErrorsWithStack("glBlitFramebuffer");
 }
 
 //------------------------------------------------------------------------------
