diff --git a/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx b/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
index 519c327b..c297f636 100644
--- a/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
+++ b/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx
@@ -1362,9 +1362,9 @@ void vtkOpenGLRenderWindow::Frame()
       {
         this->BlitDisplayFramebuffersToHardware();
       }
-      if (this->FrameBlitMode == BlitToCurrent)
+      if (this->FrameBlitMode == BlitToCurrent || this->FrameBlitMode == BlitToCurrentWithDepth)
       {
-        this->BlitDisplayFramebuffer();
+        this->BlitDisplayFramebuffer(this->FrameBlitMode == BlitToCurrentWithDepth);
       }
     }
   }
@@ -1545,10 +1545,10 @@ void vtkOpenGLRenderWindow::BlitDisplayFramebuffersToHardware()
   this->GetState()->PopFramebufferBindings();
 }
 
-void vtkOpenGLRenderWindow::BlitDisplayFramebuffer()
+void vtkOpenGLRenderWindow::BlitDisplayFramebuffer(bool includeDepth)
 {
   this->BlitDisplayFramebuffer(0, 0, 0, this->Size[0], this->Size[1], 0, 0, this->Size[0],
-    this->Size[1], GL_COLOR_BUFFER_BIT, GL_NEAREST);
+    this->Size[1], GL_COLOR_BUFFER_BIT | (includeDepth ? GL_DEPTH_BUFFER_BIT : 0), GL_NEAREST);
 }
 
 void vtkOpenGLRenderWindow::BlitDisplayFramebuffer(int right, int srcX, int srcY, int srcWidth,
diff --git a/Rendering/OpenGL2/vtkOpenGLRenderWindow.h b/Rendering/OpenGL2/vtkOpenGLRenderWindow.h
index 3fa6ae50..246140a8 100644
--- a/Rendering/OpenGL2/vtkOpenGLRenderWindow.h
+++ b/Rendering/OpenGL2/vtkOpenGLRenderWindow.h
@@ -368,7 +368,7 @@ public:
   /**
    * Blit a display framebuffer into a currently bound draw destination
    */
-  void BlitDisplayFramebuffer();
+  void BlitDisplayFramebuffer(bool includeDepth = false);
 
   /**
    * Blit a display buffer into a currently bound draw destination
@@ -393,7 +393,8 @@ public:
   enum FrameBlitModes
   {
     BlitToHardware, // hardware buffers
-    BlitToCurrent,  // currently bound draw framebuffer
+    BlitToCurrent,  // currently bound draw framebuffer, without depth buffer
+    BlitToCurrentWithDepth,  // currently bound draw framebuffer, including depth buffer
     NoBlit          // no blit, GUI or external code will handle the blit
   };
 
