diff --git a/Wrapping/PythonCore/vtkPythonArgs.cxx b/Wrapping/PythonCore/vtkPythonArgs.cxx
index 1b1e4b92..682f8b85 100644
--- a/Wrapping/PythonCore/vtkPythonArgs.cxx
+++ b/Wrapping/PythonCore/vtkPythonArgs.cxx
@@ -102,7 +102,7 @@ bool vtkPythonGetStringValue(PyObject *o, T *&a, const char *exctext)
   else if (PyUnicode_Check(o))
   {
 #if PY_VERSION_HEX >= 0x03030000
-    a = PyUnicode_AsUTF8(o);
+    a = const_cast<char*>(PyUnicode_AsUTF8(o));
     return true;
 #else
     PyObject *s = _PyUnicode_AsDefaultEncodedString(o, NULL);
