diff --git i/Rendering/Label/vtkLabelHierarchy.cxx w/Rendering/Label/vtkLabelHierarchy.cxx
index 83019c24..f856ff9b 100644
--- i/Rendering/Label/vtkLabelHierarchy.cxx
+++ w/Rendering/Label/vtkLabelHierarchy.cxx
@@ -525,7 +525,7 @@ public:
   {
   public:
     bool operator()(const vtkHierarchyNode & a,
-                    const vtkHierarchyNode & b)
+                    const vtkHierarchyNode & b) const
     {
       if (a.Level != b.Level)
       {
diff --git i/Rendering/Label/vtkLabelHierarchyPrivate.h w/Rendering/Label/vtkLabelHierarchyPrivate.h
index 557ddf0e..01135e50 100644
--- i/Rendering/Label/vtkLabelHierarchyPrivate.h
+++ w/Rendering/Label/vtkLabelHierarchyPrivate.h
@@ -66,7 +66,7 @@ public:
     {
     }
 
-    bool operator () ( const vtkIdType& a, const vtkIdType& b )
+    bool operator () ( const vtkIdType& a, const vtkIdType& b ) const
     {
       if (0 == this->Hierarchy)
       {
