diff --git i/IO/Infovis/vtkXMLTreeReader.cxx w/IO/Infovis/vtkXMLTreeReader.cxx
index 9e16a3bf..ffaf1881 100644
--- i/IO/Infovis/vtkXMLTreeReader.cxx
+++ w/IO/Infovis/vtkXMLTreeReader.cxx
@@ -221,6 +221,12 @@ int vtkXMLTreeReader::RequestData(
 
   // Get the root element node
   xmlNode *rootElement = xmlDocGetRootElement(doc);
+  if (!rootElement)
+  {
+    vtkErrorMacro(<< "Could not get root element of document.");
+    return 0;
+  }
+
   vtkXMLTreeReaderProcessElement(builder, -1, rootElement, this->ReadCharData, this->MaskArrays);
 
   xmlFreeDoc(doc);
