diff --git i/Parallel/Core/CMakeLists.txt w/Parallel/Core/CMakeLists.txt
index cb7c019e..ca23fe1c 100644
--- i/Parallel/Core/CMakeLists.txt
+++ w/Parallel/Core/CMakeLists.txt
@@ -22,7 +22,8 @@ add_custom_command(
           vtkHashSource
   COMMAND vtkHashSource
     ${VTK_SOURCE_DIR}/Parallel/Core/vtkSocketCommunicator.cxx
-    vtkSocketCommunicatorHash vtkSocketCommunicatorHash.h
+    vtkSocketCommunicatorHash ${VTK_BINARY_DIR}/Parallel/Core/vtkSocketCommunicatorHash.h
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/lib
   )
 
 list(APPEND Module_SRCS
diff --git i/Wrapping/Python/CMakeLists.txt w/Wrapping/Python/CMakeLists.txt
index 6b5c0141..4b41ec3f 100644
--- i/Wrapping/Python/CMakeLists.txt
+++ w/Wrapping/Python/CMakeLists.txt
@@ -394,6 +394,7 @@ if(PYTHON_EXECUTABLE)
         add_custom_command(DEPENDS ${src}
           COMMAND ${CMAKE_COMMAND} -E copy ${src} ${tgt}
           OUTPUT ${tgt}
+          WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
           COMMENT "source copy")
       endif ()
     endforeach()
@@ -407,6 +408,7 @@ if(PYTHON_EXECUTABLE)
     COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/compile_all_vtk.py
     DEPENDS ${VTK_PYTHON_SOURCE_FILES} ${CMAKE_CURRENT_BINARY_DIR}/compile_all_vtk.py
     OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/vtk_compile_complete"
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
     )
 
   # Create a target on which custom commands can depend.  When this
