diff --git i/Rendering/Qt/vtkQtLabelRenderStrategy.cxx w/Rendering/Qt/vtkQtLabelRenderStrategy.cxx
index 8bfd3f6b..4abfedc8 100644
--- i/Rendering/Qt/vtkQtLabelRenderStrategy.cxx
+++ w/Rendering/Qt/vtkQtLabelRenderStrategy.cxx
@@ -44,6 +44,7 @@
 #include <QPixmap>
 #include <QTextDocument>
 #include <QTextStream>
+#include <QPainterPath>
 
 vtkStandardNewMacro(vtkQtLabelRenderStrategy);
 
diff --git i/Rendering/Qt/vtkQtStringToImage.cxx w/Rendering/Qt/vtkQtStringToImage.cxx
index 1522c67e..73e5d62c 100644
--- i/Rendering/Qt/vtkQtStringToImage.cxx
+++ w/Rendering/Qt/vtkQtStringToImage.cxx
@@ -34,6 +34,7 @@
 #include <QPixmap>
 #include <QTextDocument>
 #include <QTextStream>
+#include <QPainterPath>
 
 namespace
 {
