diff --git i/CMake/VTKGenerateExportHeader.cmake w/CMake/VTKGenerateExportHeader.cmake
index 6daf9f1..6dea23d 100644
--- i/CMake/VTKGenerateExportHeader.cmake
+++ w/CMake/VTKGenerateExportHeader.cmake
@@ -183,9 +183,9 @@ macro(_vtk_test_compiler_hidden_visibility)
         _gcc_version "${_gcc_version_info}")
     endif()
 
-    if(_gcc_version VERSION_LESS "4.2")
-      set(GCC_TOO_OLD TRUE)
-    endif()
+    #if(_gcc_version VERSION_LESS "4.2")
+    #  set(GCC_TOO_OLD TRUE)
+    #endif()
   endif()
 
   if(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
