diff --git a/CMake/vtkPythonWrapping.cmake b/CMake/vtkPythonWrapping.cmake
index e695a034..8c7d1045 100644
--- a/CMake/vtkPythonWrapping.cmake
+++ b/CMake/vtkPythonWrapping.cmake
@@ -140,6 +140,9 @@ function(vtk_add_python_wrapping_library module srcs)
     set(prefix ${prefix}${VTK_KIT_SUFFIX})
   endif()
   _vtk_add_python_module(${module}Python ${prefix}PythonInit.cxx)
+  if(UNIX)
+    set_property(TARGET ${module}Python PROPERTY DEBUG_POSTFIX "")
+  endif()
   target_link_libraries(${module}Python ${module}PythonD)
   if(CMAKE_HAS_TARGET_INCLUDES)
     set_property(TARGET ${module}Python APPEND
