diff --git a/CMakeLists.txt b/CMakeLists.txt
index e1a6f54..27b7b1e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.14)

 if(COMMAND cmake_policy)
     cmake_policy(SET CMP0003 NEW)
@@ -146,9 +146,14 @@ if (BUILD_CUDA_LIB)
     endif(CUDA_FOUND)
 endif(BUILD_CUDA_LIB)

-find_package(PkgConfig REQUIRED)
-pkg_check_modules(LZ4 REQUIRED liblz4)
-include_directories(${LZ4_INCLUDE_DIRS})
+find_package(LZ4 REQUIRED)
+if(LZ4_FOUND)
+    include_directories(${LZ4_INCLUDE_DIRS})
+else()
+    message(ERROR "LZ4 NOT found")
+endif()
+
+

 #set the C/C++ include path to the "include" directory
 include_directories(BEFORE ${PROJECT_SOURCE_DIR}/src/cpp)
