diff --git i/CMakeLists.txt w/CMakeLists.txt
index e1a6f54..1d18670 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -146,9 +146,7 @@ if (BUILD_CUDA_LIB)
     endif(CUDA_FOUND)
 endif(BUILD_CUDA_LIB)
 
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(LZ4 REQUIRED liblz4)
-include_directories(${LZ4_INCLUDE_DIRS})
+set(LZ4_LINK_LIBRARIES ${CONAN_LIBS_LZ4})
 
 #set the C/C++ include path to the "include" directory
 include_directories(BEFORE ${PROJECT_SOURCE_DIR}/src/cpp)
