diff --git i/src/cpp/CMakeLists.txt w/src/cpp/CMakeLists.txt
index d8bff04..1558aa3 100644
--- i/src/cpp/CMakeLists.txt
+++ w/src/cpp/CMakeLists.txt
@@ -40,6 +40,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
         target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
         set_property(TARGET flann_cpp_s PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
     endif()
+    set_target_properties(flann_cpp PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 else()
     add_library(flann_cpp SHARED ${CPP_SOURCES})
     target_link_libraries(flann_cpp ${LZ4_LINK_LIBRARIES})
