diff --git c/src/cpp/CMakeLists.txt i/src/cpp/CMakeLists.txt
index 9c8cdbd..602a884 100644
--- c/src/cpp/CMakeLists.txt
+++ i/src/cpp/CMakeLists.txt
@@ -4,13 +4,14 @@ add_definitions(-D_FLANN_VERSION=${FLANN_VERSION})
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/flann/config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/flann/config.h)
 
-file(GLOB_RECURSE C_SOURCES flann.cpp lz4.c lz4hc.c)
-file(GLOB_RECURSE CPP_SOURCES flann_cpp.cpp lz4.c lz4hc.c)
+file(GLOB_RECURSE C_SOURCES flann.cpp)
+file(GLOB_RECURSE CPP_SOURCES flann_cpp.cpp)
 file(GLOB_RECURSE CU_SOURCES *.cu)
 
 file(TOUCH empty.cpp)
 
 add_library(flann_cpp_s STATIC ${CPP_SOURCES})
+target_link_libraries(flann_cpp_s PUBLIC ${CONAN_LIBS_LZ4})
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
     set_target_properties(flann_cpp_s PROPERTIES COMPILE_FLAGS -fPIC)
 endif()
@@ -32,6 +33,7 @@ endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
     add_library(flann_cpp SHARED empty.cpp)
+    target_link_libraries(flann_cpp PUBLIC ${CONAN_LIBS_LZ4})
     set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
     target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
 
@@ -79,6 +81,7 @@ endif()
 
 if (BUILD_C_BINDINGS)
     add_library(flann_s STATIC ${C_SOURCES})
+    target_link_libraries(flann_s PUBLIC ${CONAN_LIBS_LZ4})
     if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
         set_target_properties(flann_s PROPERTIES COMPILE_FLAGS -fPIC)
     endif()
@@ -90,6 +93,7 @@ if (BUILD_C_BINDINGS)
         target_link_libraries(flann -Wl,-whole-archive flann_s -Wl,-no-whole-archive)
     else()
         add_library(flann SHARED ${C_SOURCES})
+        target_link_libraries(flann PUBLIC ${CONAN_LIBS_LZ4})
 
         if(MINGW AND OPENMP_FOUND)
           target_link_libraries(flann gomp)
diff --git c/src/cpp/flann/util/serialization.h i/src/cpp/flann/util/serialization.h
index d731021..cdc2e5e 100644
--- c/src/cpp/flann/util/serialization.h
+++ i/src/cpp/flann/util/serialization.h
@@ -6,8 +6,8 @@
 #include <cstdlib>
 #include <cstring>
 #include <stdio.h>
-#include "flann/ext/lz4.h"
-#include "flann/ext/lz4hc.h"
+#include <lz4.h>
+#include <lz4hc.h>
 
 
 namespace flann
